// angular-loading-page/esm2015/lib/loaders/folding-cube/folding-cube.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FoldingCubeComponent } from './folding-cube.component';
import * as ɵngcc0 from '@angular/core';
export class FoldingCubeModule {
}
FoldingCubeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FoldingCubeModule });
FoldingCubeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FoldingCubeModule_Factory(t) { return new (t || FoldingCubeModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FoldingCubeModule, { declarations: function () { return [FoldingCubeComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [FoldingCubeComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FoldingCubeModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    FoldingCubeComponent,
                ],
                exports: [
                    FoldingCubeComponent
                ]
            }]
    }], null, null); })();
