// angular-loading-page/esm2015/lib/loaders/folding-cube/folding-cube.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class FoldingCubeComponent {
    constructor() {
        this.color = '#da4733';
    }
}
FoldingCubeComponent.ɵfac = function FoldingCubeComponent_Factory(t) { return new (t || FoldingCubeComponent)(); };
FoldingCubeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FoldingCubeComponent, selectors: [["mk-folding-cube"]], inputs: { color: "color" }, decls: 9, vars: 8, consts: [[1, "sk-folding-cube"], [1, "sk-cube1", "sk-cube"], [1, "sk-cube2", "sk-cube"], [1, "sk-cube4", "sk-cube"], [1, "sk-cube3", "sk-cube"]], template: function FoldingCubeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-folding-cube[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;transform:rotate(45deg);width:40px}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{float:left;height:50%;position:relative;transform:scale(1.1);width:50%}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:sk-foldCubeAngle 2.4s linear infinite both;content:\"\";height:100%;left:0;position:absolute;top:0;transform-origin:100% 100%;width:100%}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{transform:scale(1.1) rotate(90deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%]{transform:scale(1.1) rotate(180deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%]{transform:scale(1.1) rotate(270deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:.3s}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:.6s}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:.9s}@keyframes sk-foldCubeAngle{0%,10%{opacity:0;transform:perspective(140px) rotateX(-180deg)}25%,75%{opacity:1;transform:perspective(140px) rotateX(0deg)}90%,to{opacity:0;transform:perspective(140px) rotateY(180deg)}}"] });
FoldingCubeComponent.propDecorators = {
    color: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FoldingCubeComponent, [{
        type: Component,
        args: [{
                selector: 'mk-folding-cube',
                template: `
    <div class="sk-folding-cube">
      <div class="sk-cube1 sk-cube"><div [style.background-color]="color"></div></div>
      <div class="sk-cube2 sk-cube"><div [style.background-color]="color"></div></div>
      <div class="sk-cube4 sk-cube"><div [style.background-color]="color"></div></div>
      <div class="sk-cube3 sk-cube"><div [style.background-color]="color"></div></div>
    </div>`,
                styles: [".sk-folding-cube{height:40px;margin:40px auto;position:relative;transform:rotate(45deg);width:40px}.sk-folding-cube .sk-cube{float:left;height:50%;position:relative;transform:scale(1.1);width:50%}.sk-folding-cube .sk-cube>div{animation:sk-foldCubeAngle 2.4s linear infinite both;content:\"\";height:100%;left:0;position:absolute;top:0;transform-origin:100% 100%;width:100%}.sk-folding-cube .sk-cube2{transform:scale(1.1) rotate(90deg)}.sk-folding-cube .sk-cube3{transform:scale(1.1) rotate(180deg)}.sk-folding-cube .sk-cube4{transform:scale(1.1) rotate(270deg)}.sk-folding-cube .sk-cube2>div{animation-delay:.3s}.sk-folding-cube .sk-cube3>div{animation-delay:.6s}.sk-folding-cube .sk-cube4>div{animation-delay:.9s}@keyframes sk-foldCubeAngle{0%,10%{opacity:0;transform:perspective(140px) rotateX(-180deg)}25%,75%{opacity:1;transform:perspective(140px) rotateX(0deg)}90%,to{opacity:0;transform:perspective(140px) rotateY(180deg)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
