// angular-loading-page/esm2015/lib/loaders/three-bounce/three-bounce.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ThreeBounceComponent } from './three-bounce.component';
import * as ɵngcc0 from '@angular/core';
export class ThreeBounceModule {
}
ThreeBounceModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ThreeBounceModule });
ThreeBounceModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ThreeBounceModule_Factory(t) { return new (t || ThreeBounceModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ThreeBounceModule, { declarations: function () { return [ThreeBounceComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ThreeBounceComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ThreeBounceModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    ThreeBounceComponent,
                ],
                exports: [
                    ThreeBounceComponent
                ]
            }]
    }], null, null); })();
