// angular-loading-page/esm2015/lib/loading-page.module.js
import { NgModule, Injector } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { RouteInterceptorService } from './interceptors/route-interceptor.service';
import { HttpInterceptorService, HttpInterceptorServiceFactory } from './interceptors/http-interceptor.service';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpObservableService } from './interceptors/http-observable.service';
import { LoadingPageDirective } from './loading-page.directive';
import * as ɵngcc0 from '@angular/core';
const HttpInterceptorServiceExistingProvider = {
    provide: HTTP_INTERCEPTORS,
    useExisting: HttpInterceptorService,
    multi: true
};
const ɵ0 = HttpInterceptorServiceFactory;
export class LoadingPageModule {
}
LoadingPageModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LoadingPageModule });
LoadingPageModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function LoadingPageModule_Factory(t) { return new (t || LoadingPageModule)(); }, providers: [
        RouteInterceptorService,
        HttpObservableService,
        HttpInterceptorServiceExistingProvider,
        {
            provide: HttpInterceptorService,
            useFactory: ɵ0,
            deps: [Injector]
        }
    ], imports: [[
            CommonModule,
            HttpClientModule
        ], HttpClientModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LoadingPageModule, { declarations: function () { return [LoadingPageDirective]; }, imports: function () { return [CommonModule,
        HttpClientModule]; }, exports: function () { return [HttpClientModule,
        LoadingPageDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoadingPageModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    HttpClientModule
                ],
                declarations: [
                    LoadingPageDirective
                ],
                exports: [
                    HttpClientModule,
                    LoadingPageDirective
                ],
                providers: [
                    RouteInterceptorService,
                    HttpObservableService,
                    HttpInterceptorServiceExistingProvider,
                    {
                        provide: HttpInterceptorService,
                        useFactory: ɵ0,
                        deps: [Injector]
                    }
                ]
            }]
    }], null, null); })();
export { ɵ0 };
