// angular-loading-page/esm2015/lib/loading-page.module.js
import { NgModule, Injector } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClientModule } from '@angular/common/http';
import { RouteInterceptorService } from './interceptors/route-interceptor.service';
import { HttpInterceptorService, HttpInterceptorServiceFactory } from './interceptors/http-interceptor.service';
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { HttpObservableService } from './interceptors/http-observable.service';
import { LoadingPageDirective } from './loading-page.directive';
const HttpInterceptorServiceExistingProvider = {
    provide: HTTP_INTERCEPTORS,
    useExisting: HttpInterceptorService,
    multi: true
};
const ɵ0 = HttpInterceptorServiceFactory;
export class LoadingPageModule {
}
LoadingPageModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule,
                    HttpClientModule
                ],
                declarations: [
                    LoadingPageDirective
                ],
                exports: [
                    HttpClientModule,
                    LoadingPageDirective
                ],
                providers: [
                    RouteInterceptorService,
                    HttpObservableService,
                    HttpInterceptorServiceExistingProvider,
                    {
                        provide: HttpInterceptorService,
                        useFactory: ɵ0,
                        deps: [Injector]
                    }
                ]
            },] }
];
export { ɵ0 };
