// angular-admin-lte/esm2015/lib/layout/wrapper/wrapper.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WrapperComponent } from './wrapper.component';
import * as ɵngcc0 from '@angular/core';
export { WrapperComponent } from './wrapper.component';
export class WrapperModule {
}
WrapperModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WrapperModule });
WrapperModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WrapperModule_Factory(t) { return new (t || WrapperModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WrapperModule, { declarations: function () { return [WrapperComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [WrapperComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WrapperModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [WrapperComponent],
                declarations: [WrapperComponent]
            }]
    }], null, null); })();
