// angular-loading-page/esm2015/lib/interceptors/http-interceptor.service.js
import { Injectable, Injector } from '@angular/core';
import { HttpObservableService } from './http-observable.service';
import { finalize } from 'rxjs/operators';
import * as ɵngcc0 from '@angular/core';
export class HttpInterceptorService {
    /**
     * @method constructor
     * @param injector [description]
     */
    constructor(injector) {
        this.injector = injector;
        this.httpObservableService = injector.get(HttpObservableService);
    }
    /**
     * @method intercept
     * @param request [description]
     * @param next [description]
     */
    intercept(request, next) {
        this.httpObservableService.onRequestStart();
        return next.handle(request).pipe(finalize(() => {
            this.httpObservableService.onRequestDone();
        }));
    }
}
HttpInterceptorService.ɵfac = function HttpInterceptorService_Factory(t) { return new (t || HttpInterceptorService)(ɵngcc0.ɵɵinject(ɵngcc0.Injector)); };
HttpInterceptorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: HttpInterceptorService, factory: HttpInterceptorService.ɵfac });
HttpInterceptorService.ctorParameters = () => [
    { type: Injector }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HttpInterceptorService, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc0.Injector }]; }, null); })();
export function HttpInterceptorServiceFactory(injector) {
    return new HttpInterceptorService(injector);
}
