// angular-loading-page/esm2015/lib/interceptors/http-observable.service.js
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
export class HttpObservableService {
    constructor() {
        this.pendingRequestsCounter = 0;
        this.isPendingRequestsSubject = new Subject();
    }
    /**
     * [isPendingRequests description]
     * @method isPendingRequests
     * @return [description]
     */
    get isPendingRequests() {
        return this.isPendingRequestsSubject.asObservable();
    }
    /**
     * [pendingRequests description]
     * @method pendingRequests
     * @return [description]
     */
    get pendingRequests() {
        return this.pendingRequestsCounter;
    }
    /**
     * [onRequestStart description]
     * @method onRequestStart
     */
    onRequestStart() {
        this.pendingRequestsCounter++;
        if (this.pendingRequestsCounter === 1) {
            this.isPendingRequestsSubject.next(true);
        }
    }
    /**
     * [onRequestDone description]
     * @method onRequestDone
     */
    onRequestDone() {
        this.pendingRequestsCounter--;
        if (this.pendingRequestsCounter === 0) {
            this.isPendingRequestsSubject.next(false);
        }
    }
}
HttpObservableService.ɵfac = function HttpObservableService_Factory(t) { return new (t || HttpObservableService)(); };
HttpObservableService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: HttpObservableService, factory: HttpObservableService.ɵfac });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HttpObservableService, [{
        type: Injectable
    }], function () { return []; }, null); })();
