// angular-loading-page/esm2015/lib/interceptors/route-interceptor.service.js
import { Injectable, Optional } from '@angular/core';
import { Router, NavigationStart, NavigationEnd, NavigationCancel, NavigationError } from '@angular/router';
import { Subject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
export class RouteInterceptorService {
    /**
     * @method constructor
     */
    constructor(router) {
        this.router = router;
        this.isPendingRouteSubject = new Subject();
        if (this.router) {
            router.events.subscribe((event) => {
                this.routerInterceptor(event);
            });
        }
    }
    /**
     * [isPendingRoute description]
     * @method isPendingRoute
     * @return [description]
     */
    get isPendingRoute() {
        return this.isPendingRouteSubject.asObservable();
    }
    /**
     * [routerInterceptor description]
     * @method routerInterceptor
     * @param event [description]
     */
    routerInterceptor(event) {
        if (event instanceof NavigationStart) {
            this.onRouteStart();
        }
        if (event instanceof NavigationEnd || event instanceof NavigationCancel || event instanceof NavigationError) {
            this.onRouteDone();
        }
    }
    /**
     * [onRequestStart description]
     * @method onRequestStart
     */
    onRouteStart() {
        this.isPendingRouteSubject.next(true);
    }
    /**
     * [onRequestDone description]
     * @method onRequestDone
     */
    onRouteDone() {
        this.isPendingRouteSubject.next(false);
    }
}
RouteInterceptorService.ɵfac = function RouteInterceptorService_Factory(t) { return new (t || RouteInterceptorService)(ɵngcc0.ɵɵinject(ɵngcc1.Router, 8)); };
RouteInterceptorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: RouteInterceptorService, factory: RouteInterceptorService.ɵfac });
RouteInterceptorService.ctorParameters = () => [
    { type: Router, decorators: [{ type: Optional }] }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RouteInterceptorService, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.Router, decorators: [{
                type: Optional
            }] }]; }, null); })();
