// angular-admin-lte/esm2015/lib/layout/sidebar-left/sidebar-left.directive.js
import { Directive, Input, ElementRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class SidebarLeftToggleDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
    }
}
SidebarLeftToggleDirective.ɵfac = function SidebarLeftToggleDirective_Factory(t) { return new (t || SidebarLeftToggleDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
SidebarLeftToggleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SidebarLeftToggleDirective, selectors: [["", "mkMenuToggle", ""]], inputs: { item: ["mkMenuToggle", "item"] } });
SidebarLeftToggleDirective.ctorParameters = () => [
    { type: ElementRef }
];
SidebarLeftToggleDirective.propDecorators = {
    item: [{ type: Input, args: ['mkMenuToggle',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarLeftToggleDirective, [{
        type: Directive,
        args: [{
                selector: '[mkMenuToggle]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { item: [{
            type: Input,
            args: ['mkMenuToggle']
        }] }); })();
