// angular-admin-lte/esm2015/lib/layout/sidebar-left/sidebar-left.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { AnimationsModule } from '../../animations/animations.module';
import { SidebarLeftToggleDirective } from './sidebar-left.directive';
import { SidebarLeftComponent } from './sidebar-left.component';
import * as ɵngcc0 from '@angular/core';
export { SidebarLeftToggleDirective } from './sidebar-left.directive';
export { SidebarLeftComponent } from './sidebar-left.component';
export class SidebarLeftModule {
}
SidebarLeftModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SidebarLeftModule });
SidebarLeftModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SidebarLeftModule_Factory(t) { return new (t || SidebarLeftModule)(); }, imports: [[CommonModule, RouterModule, AnimationsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SidebarLeftModule, { declarations: function () { return [SidebarLeftToggleDirective, SidebarLeftComponent]; }, imports: function () { return [CommonModule, RouterModule, AnimationsModule]; }, exports: function () { return [SidebarLeftComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarLeftModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule, AnimationsModule],
                exports: [SidebarLeftComponent],
                declarations: [SidebarLeftToggleDirective, SidebarLeftComponent]
            }]
    }], null, null); })();
