// angular-loading-page/esm2015/lib/loaders/pulse/pulse.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PulseComponent } from './pulse.component';
import * as ɵngcc0 from '@angular/core';
export class PulseModule {
}
PulseModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: PulseModule });
PulseModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function PulseModule_Factory(t) { return new (t || PulseModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(PulseModule, { declarations: function () { return [PulseComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [PulseComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PulseModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    PulseComponent,
                ],
                exports: [
                    PulseComponent
                ]
            }]
    }], null, null); })();
