// angular-loading-page/esm2015/lib/loaders/pulse/pulse.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class PulseComponent {
    constructor() {
        this.color = '#da4733';
    }
}
PulseComponent.ɵfac = function PulseComponent_Factory(t) { return new (t || PulseComponent)(); };
PulseComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PulseComponent, selectors: [["mk-pulse"]], inputs: { color: "color" }, decls: 1, vars: 2, consts: [[1, "sk-spinner", "sk-spinner-pulse"]], template: function PulseComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-spinner-pulse[_ngcontent-%COMP%]{animation:sk-pulseScaleOut 1s ease-in-out infinite;background-color:#333;border-radius:100%;height:40px;margin:40px auto;width:40px}@keyframes sk-pulseScaleOut{0%{transform:scale(0)}to{opacity:0;transform:scale(1)}}"] });
PulseComponent.propDecorators = {
    color: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PulseComponent, [{
        type: Component,
        args: [{
                selector: 'mk-pulse',
                template: `<div class="sk-spinner sk-spinner-pulse" [style.background-color]="color"></div>`,
                styles: [".sk-spinner-pulse{animation:sk-pulseScaleOut 1s ease-in-out infinite;background-color:#333;border-radius:100%;height:40px;margin:40px auto;width:40px}@keyframes sk-pulseScaleOut{0%{transform:scale(0)}to{opacity:0;transform:scale(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
