// angular-loading-page/esm2015/lib/loaders/cube-grid/cube-grid.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CubeGridComponent } from './cube-grid.component';
import * as ɵngcc0 from '@angular/core';
export class CubeGridModule {
}
CubeGridModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CubeGridModule });
CubeGridModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CubeGridModule_Factory(t) { return new (t || CubeGridModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CubeGridModule, { declarations: function () { return [CubeGridComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CubeGridComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CubeGridModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    CubeGridComponent,
                ],
                exports: [
                    CubeGridComponent
                ]
            }]
    }], null, null); })();
