// angular-loading-page/esm2015/lib/loaders/cube-grid/cube-grid.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class CubeGridComponent {
    constructor() {
        this.color = '#da4733';
    }
}
CubeGridComponent.ɵfac = function CubeGridComponent_Factory(t) { return new (t || CubeGridComponent)(); };
CubeGridComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CubeGridComponent, selectors: [["mk-cube-grid"]], inputs: { color: "color" }, decls: 10, vars: 18, consts: [[1, "sk-cube-grid"], [1, "sk-cube", "sk-cube1"], [1, "sk-cube", "sk-cube2"], [1, "sk-cube", "sk-cube3"], [1, "sk-cube", "sk-cube4"], [1, "sk-cube", "sk-cube5"], [1, "sk-cube", "sk-cube6"], [1, "sk-cube", "sk-cube7"], [1, "sk-cube", "sk-cube8"], [1, "sk-cube", "sk-cube9"]], template: function CubeGridComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelement(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelement(6, "div", 6);
        ɵngcc0.ɵɵelement(7, "div", 7);
        ɵngcc0.ɵɵelement(8, "div", 8);
        ɵngcc0.ɵɵelement(9, "div", 9);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-cube-grid[_ngcontent-%COMP%]{height:40px;margin:40px auto;width:40px}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{animation:sk-cubeGridScaleDelay 1.3s ease-in-out infinite;float:left;height:33.33%;width:33.33%}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube1[_ngcontent-%COMP%]{animation-delay:.2s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{animation-delay:.3s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%]{animation-delay:.4s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%]{animation-delay:.1s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube5[_ngcontent-%COMP%]{animation-delay:.2s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube6[_ngcontent-%COMP%]{animation-delay:.3s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube7[_ngcontent-%COMP%]{animation-delay:0s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube8[_ngcontent-%COMP%]{animation-delay:.1s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube9[_ngcontent-%COMP%]{animation-delay:.2s}@keyframes sk-cubeGridScaleDelay{0%,70%,to{transform:scaleX(1)}35%{transform:scale3D(0,0,1)}}"] });
CubeGridComponent.propDecorators = {
    color: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CubeGridComponent, [{
        type: Component,
        args: [{
                selector: 'mk-cube-grid',
                template: `
    <div class="sk-cube-grid">
      <div class="sk-cube sk-cube1" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube2" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube3" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube4" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube5" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube6" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube7" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube8" [style.background-color]="color"></div>
      <div class="sk-cube sk-cube9" [style.background-color]="color"></div>
    </div>`,
                styles: [".sk-cube-grid{height:40px;margin:40px auto;width:40px}.sk-cube-grid .sk-cube{animation:sk-cubeGridScaleDelay 1.3s ease-in-out infinite;float:left;height:33.33%;width:33.33%}.sk-cube-grid .sk-cube1{animation-delay:.2s}.sk-cube-grid .sk-cube2{animation-delay:.3s}.sk-cube-grid .sk-cube3{animation-delay:.4s}.sk-cube-grid .sk-cube4{animation-delay:.1s}.sk-cube-grid .sk-cube5{animation-delay:.2s}.sk-cube-grid .sk-cube6{animation-delay:.3s}.sk-cube-grid .sk-cube7{animation-delay:0s}.sk-cube-grid .sk-cube8{animation-delay:.1s}.sk-cube-grid .sk-cube9{animation-delay:.2s}@keyframes sk-cubeGridScaleDelay{0%,70%,to{transform:scaleX(1)}35%{transform:scale3D(0,0,1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
