// angular-loading-page/esm2015/lib/loaders/wandering-cubes/wandering-cubes.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WanderingCubesComponent } from './wandering-cubes.component';
import * as ɵngcc0 from '@angular/core';
export class WanderingCubesModule {
}
WanderingCubesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WanderingCubesModule });
WanderingCubesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WanderingCubesModule_Factory(t) { return new (t || WanderingCubesModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WanderingCubesModule, { declarations: function () { return [WanderingCubesComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [WanderingCubesComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WanderingCubesModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    WanderingCubesComponent,
                ],
                exports: [
                    WanderingCubesComponent
                ]
            }]
    }], null, null); })();
