// angular-loading-page/esm2015/lib/loaders/wandering-cubes/wandering-cubes.component.js
import { Component, Input } from '@angular/core';
export class WanderingCubesComponent {
    constructor() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
}
WanderingCubesComponent.decorators = [
    { type: Component, args: [{
                selector: 'mk-wandering-cubes',
                template: `
    <div class="sk-wandering-cubes">
      <div class="sk-cube sk-cube1" [style.background-color]="colorOne"></div>
      <div class="sk-cube sk-cube2" [style.background-color]="colorTwo"></div>
    </div>`,
                styles: [".sk-wandering-cubes{height:40px;margin:40px auto;position:relative;width:40px}.sk-wandering-cubes .sk-cube{animation:sk-wanderingCube 1.8s ease-in-out -1.8s infinite both;height:10px;left:0;position:absolute;top:0;width:10px}.sk-wandering-cubes .sk-cube2{animation-delay:-.9s}@keyframes sk-wanderingCube{0%{transform:rotate(0deg)}25%{transform:translateX(30px) rotate(-90deg) scale(.5)}50%{transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}to{transform:rotate(-1turn)}}"]
            },] }
];
WanderingCubesComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }]
};
