// angular-admin-lte/esm2015/lib/color/color.module.js
import { NgModule } from '@angular/core';
import { BackgroundColorDirective, ColorDirective } from './color.directive';
import * as ɵngcc0 from '@angular/core';
export { BackgroundColorDirective, ColorDirective } from './color.directive';
export class ColorModule {
}
ColorModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ColorModule });
ColorModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ColorModule_Factory(t) { return new (t || ColorModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ColorModule, { declarations: function () { return [BackgroundColorDirective, ColorDirective]; }, exports: function () { return [BackgroundColorDirective, ColorDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ColorModule, [{
        type: NgModule,
        args: [{
                exports: [BackgroundColorDirective, ColorDirective],
                declarations: [BackgroundColorDirective, ColorDirective]
            }]
    }], null, null); })();
