// angular-loading-page/esm2015/lib/loaders/chasing-dots/chasing-dots.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ChasingDotsComponent } from './chasing-dots.component';
import * as ɵngcc0 from '@angular/core';
export class ChasingDotsModule {
}
ChasingDotsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChasingDotsModule });
ChasingDotsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChasingDotsModule_Factory(t) { return new (t || ChasingDotsModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChasingDotsModule, { declarations: function () { return [ChasingDotsComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ChasingDotsComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChasingDotsModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    ChasingDotsComponent,
                ],
                exports: [
                    ChasingDotsComponent
                ]
            }]
    }], null, null); })();
