// angular-loading-page/esm2015/lib/loaders/chasing-dots/chasing-dots.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class ChasingDotsComponent {
    constructor() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
}
ChasingDotsComponent.ɵfac = function ChasingDotsComponent_Factory(t) { return new (t || ChasingDotsComponent)(); };
ChasingDotsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ChasingDotsComponent, selectors: [["mk-chasing-dots"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-chasing-dots"], [1, "sk-child", "sk-dot1"], [1, "sk-child", "sk-dot2"]], template: function ChasingDotsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-chasing-dots[_ngcontent-%COMP%]{animation:sk-chasingDotsRotate 2s linear infinite;height:40px;margin:40px auto;position:relative;text-align:center;width:40px}.sk-chasing-dots[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{animation:sk-chasingDotsBounce 2s ease-in-out infinite;border-radius:100%;display:inline-block;height:60%;position:absolute;top:0;width:60%}.sk-chasing-dots[_ngcontent-%COMP%]   .sk-dot2[_ngcontent-%COMP%]{animation-delay:-1s;bottom:0;top:auto}@keyframes sk-chasingDotsRotate{to{transform:rotate(1turn)}}@keyframes sk-chasingDotsBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"] });
ChasingDotsComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChasingDotsComponent, [{
        type: Component,
        args: [{
                selector: 'mk-chasing-dots',
                template: `
    <div class="sk-chasing-dots">
      <div class="sk-child sk-dot1" [style.background-color]="colorOne"></div>
      <div class="sk-child sk-dot2" [style.background-color]="colorTwo"></div>
    </div>`,
                styles: [".sk-chasing-dots{animation:sk-chasingDotsRotate 2s linear infinite;height:40px;margin:40px auto;position:relative;text-align:center;width:40px}.sk-chasing-dots .sk-child{animation:sk-chasingDotsBounce 2s ease-in-out infinite;border-radius:100%;display:inline-block;height:60%;position:absolute;top:0;width:60%}.sk-chasing-dots .sk-dot2{animation-delay:-1s;bottom:0;top:auto}@keyframes sk-chasingDotsRotate{to{transform:rotate(1turn)}}@keyframes sk-chasingDotsBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
