// angular-loading-page/esm2015/lib/loaders/chasing-dots/chasing-dots.component.js
import { Component, Input } from '@angular/core';
export class ChasingDotsComponent {
    constructor() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
}
ChasingDotsComponent.decorators = [
    { type: Component, args: [{
                selector: 'mk-chasing-dots',
                template: `
    <div class="sk-chasing-dots">
      <div class="sk-child sk-dot1" [style.background-color]="colorOne"></div>
      <div class="sk-child sk-dot2" [style.background-color]="colorTwo"></div>
    </div>`,
                styles: [".sk-chasing-dots{animation:sk-chasingDotsRotate 2s linear infinite;height:40px;margin:40px auto;position:relative;text-align:center;width:40px}.sk-chasing-dots .sk-child{animation:sk-chasingDotsBounce 2s ease-in-out infinite;border-radius:100%;display:inline-block;height:60%;position:absolute;top:0;width:60%}.sk-chasing-dots .sk-dot2{animation-delay:-1s;bottom:0;top:auto}@keyframes sk-chasingDotsRotate{to{transform:rotate(1turn)}}@keyframes sk-chasingDotsBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
            },] }
];
ChasingDotsComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }]
};
