// angular-loading-page/bundles/angular-loading-page.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/common/http'), require('@angular/router'), require('rxjs'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('angular-loading-page', ['exports', '@angular/core', '@angular/common', '@angular/common/http', '@angular/router', 'rxjs', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global['angular-loading-page'] = {}, global.ng.core, global.ng.common, global.ng.common.http, global.ng.router, global.rxjs, global.rxjs.operators));
}(this, (function (exports, core, common, http, router, rxjs, operators) { 'use strict';

    var RouteInterceptorService = /** @class */ (function () {
        /**
         * @method constructor
         */
        function RouteInterceptorService(router) {
            var _this = this;
            this.router = router;
            this.isPendingRouteSubject = new rxjs.Subject();
            if (this.router) {
                router.events.subscribe(function (event) {
                    _this.routerInterceptor(event);
                });
            }
        }
        Object.defineProperty(RouteInterceptorService.prototype, "isPendingRoute", {
            /**
             * [isPendingRoute description]
             * @method isPendingRoute
             * @return [description]
             */
            get: function () {
                return this.isPendingRouteSubject.asObservable();
            },
            enumerable: false,
            configurable: true
        });
        /**
         * [routerInterceptor description]
         * @method routerInterceptor
         * @param event [description]
         */
        RouteInterceptorService.prototype.routerInterceptor = function (event) {
            if (event instanceof router.NavigationStart) {
                this.onRouteStart();
            }
            if (event instanceof router.NavigationEnd || event instanceof router.NavigationCancel || event instanceof router.NavigationError) {
                this.onRouteDone();
            }
        };
        /**
         * [onRequestStart description]
         * @method onRequestStart
         */
        RouteInterceptorService.prototype.onRouteStart = function () {
            this.isPendingRouteSubject.next(true);
        };
        /**
         * [onRequestDone description]
         * @method onRequestDone
         */
        RouteInterceptorService.prototype.onRouteDone = function () {
            this.isPendingRouteSubject.next(false);
        };
        return RouteInterceptorService;
    }());
    RouteInterceptorService.decorators = [
        { type: core.Injectable }
    ];
    RouteInterceptorService.ctorParameters = function () { return [
        { type: router.Router, decorators: [{ type: core.Optional }] }
    ]; };

    var HttpObservableService = /** @class */ (function () {
        function HttpObservableService() {
            this.pendingRequestsCounter = 0;
            this.isPendingRequestsSubject = new rxjs.Subject();
        }
        Object.defineProperty(HttpObservableService.prototype, "isPendingRequests", {
            /**
             * [isPendingRequests description]
             * @method isPendingRequests
             * @return [description]
             */
            get: function () {
                return this.isPendingRequestsSubject.asObservable();
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(HttpObservableService.prototype, "pendingRequests", {
            /**
             * [pendingRequests description]
             * @method pendingRequests
             * @return [description]
             */
            get: function () {
                return this.pendingRequestsCounter;
            },
            enumerable: false,
            configurable: true
        });
        /**
         * [onRequestStart description]
         * @method onRequestStart
         */
        HttpObservableService.prototype.onRequestStart = function () {
            this.pendingRequestsCounter++;
            if (this.pendingRequestsCounter === 1) {
                this.isPendingRequestsSubject.next(true);
            }
        };
        /**
         * [onRequestDone description]
         * @method onRequestDone
         */
        HttpObservableService.prototype.onRequestDone = function () {
            this.pendingRequestsCounter--;
            if (this.pendingRequestsCounter === 0) {
                this.isPendingRequestsSubject.next(false);
            }
        };
        return HttpObservableService;
    }());
    HttpObservableService.decorators = [
        { type: core.Injectable }
    ];

    var HttpInterceptorService = /** @class */ (function () {
        /**
         * @method constructor
         * @param injector [description]
         */
        function HttpInterceptorService(injector) {
            this.injector = injector;
            this.httpObservableService = injector.get(HttpObservableService);
        }
        /**
         * @method intercept
         * @param request [description]
         * @param next [description]
         */
        HttpInterceptorService.prototype.intercept = function (request, next) {
            var _this = this;
            this.httpObservableService.onRequestStart();
            return next.handle(request).pipe(operators.finalize(function () {
                _this.httpObservableService.onRequestDone();
            }));
        };
        return HttpInterceptorService;
    }());
    HttpInterceptorService.decorators = [
        { type: core.Injectable }
    ];
    HttpInterceptorService.ctorParameters = function () { return [
        { type: core.Injector }
    ]; };
    function HttpInterceptorServiceFactory(injector) {
        return new HttpInterceptorService(injector);
    }

    /*
     * mkLoadingPage Directive
     */
    var LoadingPageDirective = /** @class */ (function () {
        /**
         * @method constructor
         * @param privatehttpObservableService   [description]
         * @param privaterouteInterceptorService [description]
         * @param privatetemplateRef             [description]
         * @param privateviewContainer           [description]                                              [description]
         */
        function LoadingPageDirective(httpObservableService, routeInterceptorService, templateRef, viewContainer) {
            this.httpObservableService = httpObservableService;
            this.routeInterceptorService = routeInterceptorService;
            this.templateRef = templateRef;
            this.viewContainer = viewContainer;
            this.isPendingRequests = false;
            this.isPendingRoute = false;
            this.hasView = false;
        }
        /**
         * @method ngOnChanges
         * @param changes [description]
         */
        LoadingPageDirective.prototype.ngOnChanges = function (changes) {
            if (changes.mkLoadingPage.currentValue) {
                this.mkLoadingPage = changes.mkLoadingPage.currentValue;
            }
            if (changes.mkLoadingPage.firstChange) {
                this.init();
            }
        };
        /**
         * [init description]
         * @method init
         * @return [description]
         */
        LoadingPageDirective.prototype.init = function () {
            var _this = this;
            if (!this.mkLoadingPage || this.mkLoadingPage.checkPendingHttp) {
                this.httpObservableService.isPendingRequests.subscribe(function (value) {
                    _this.isPendingRequests = value;
                    _this.isVisible();
                });
            }
            if (!this.mkLoadingPage || this.mkLoadingPage.checkPendingRoute) {
                this.routeInterceptorService.isPendingRoute.subscribe(function (value) {
                    _this.isPendingRoute = value;
                    _this.isVisible();
                });
            }
        };
        /**
         * [isVisible description]
         * @method isVisible
         * @return [description]
         */
        LoadingPageDirective.prototype.isVisible = function () {
            if (this.hasView && !this.isPendingRequests && !this.isPendingRoute) {
                this.viewContainer.clear();
                this.hasView = false;
            }
            else if (!this.hasView && (this.isPendingRequests || this.isPendingRoute)) {
                this.viewContainer.createEmbeddedView(this.templateRef);
                this.hasView = true;
            }
        };
        return LoadingPageDirective;
    }());
    LoadingPageDirective.decorators = [
        { type: core.Directive, args: [{
                    selector: '[mkLoadingPage]',
                },] }
    ];
    LoadingPageDirective.ctorParameters = function () { return [
        { type: HttpObservableService },
        { type: RouteInterceptorService },
        { type: core.TemplateRef },
        { type: core.ViewContainerRef }
    ]; };
    LoadingPageDirective.propDecorators = {
        mkLoadingPage: [{ type: core.Input }]
    };

    var HttpInterceptorServiceExistingProvider = {
        provide: http.HTTP_INTERCEPTORS,
        useExisting: HttpInterceptorService,
        multi: true
    };
    var ɵ0 = HttpInterceptorServiceFactory;
    var LoadingPageModule = /** @class */ (function () {
        function LoadingPageModule() {
        }
        return LoadingPageModule;
    }());
    LoadingPageModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule,
                        http.HttpClientModule
                    ],
                    declarations: [
                        LoadingPageDirective
                    ],
                    exports: [
                        http.HttpClientModule,
                        LoadingPageDirective
                    ],
                    providers: [
                        RouteInterceptorService,
                        HttpObservableService,
                        HttpInterceptorServiceExistingProvider,
                        {
                            provide: HttpInterceptorService,
                            useFactory: ɵ0,
                            deps: [core.Injector]
                        }
                    ]
                },] }
    ];

    var ChasingDotsComponent = /** @class */ (function () {
        function ChasingDotsComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
        }
        return ChasingDotsComponent;
    }());
    ChasingDotsComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-chasing-dots',
                    template: "\n    <div class=\"sk-chasing-dots\">\n      <div class=\"sk-child sk-dot1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-child sk-dot2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                    styles: [".sk-chasing-dots{animation:sk-chasingDotsRotate 2s linear infinite;height:40px;margin:40px auto;position:relative;text-align:center;width:40px}.sk-chasing-dots .sk-child{animation:sk-chasingDotsBounce 2s ease-in-out infinite;border-radius:100%;display:inline-block;height:60%;position:absolute;top:0;width:60%}.sk-chasing-dots .sk-dot2{animation-delay:-1s;bottom:0;top:auto}@keyframes sk-chasingDotsRotate{to{transform:rotate(1turn)}}@keyframes sk-chasingDotsBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
                },] }
    ];
    ChasingDotsComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var ChasingDotsModule = /** @class */ (function () {
        function ChasingDotsModule() {
        }
        return ChasingDotsModule;
    }());
    ChasingDotsModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        ChasingDotsComponent,
                    ],
                    exports: [
                        ChasingDotsComponent
                    ]
                },] }
    ];

    var CircleComponent = /** @class */ (function () {
        function CircleComponent() {
            this.color = '#da4733';
        }
        return CircleComponent;
    }());
    CircleComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-circle',
                    template: "\n    <div class=\"sk-circle\">\n      <div class=\"sk-circle1 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle2 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle3 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle4 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle5 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle6 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle7 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle8 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle9 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle10 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle11 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle12 sk-child\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                    styles: [".sk-circle{height:40px;margin:40px auto;position:relative;width:40px}.sk-circle .sk-child{height:100%;left:0;position:absolute;top:0;width:100%}.sk-circle .sk-child>div{animation:sk-circleBounceDelay 1.2s ease-in-out infinite both;border-radius:100%;display:block;height:15%;margin:0 auto;width:15%}.sk-circle .sk-circle2{transform:rotate(30deg)}.sk-circle .sk-circle3{transform:rotate(60deg)}.sk-circle .sk-circle4{transform:rotate(90deg)}.sk-circle .sk-circle5{transform:rotate(120deg)}.sk-circle .sk-circle6{transform:rotate(150deg)}.sk-circle .sk-circle7{transform:rotate(180deg)}.sk-circle .sk-circle8{transform:rotate(210deg)}.sk-circle .sk-circle9{transform:rotate(240deg)}.sk-circle .sk-circle10{transform:rotate(270deg)}.sk-circle .sk-circle11{transform:rotate(300deg)}.sk-circle .sk-circle12{transform:rotate(330deg)}.sk-circle .sk-circle2>div{animation-delay:-1.1s}.sk-circle .sk-circle3>div{animation-delay:-1s}.sk-circle .sk-circle4>div{animation-delay:-.9s}.sk-circle .sk-circle5>div{animation-delay:-.8s}.sk-circle .sk-circle6>div{animation-delay:-.7s}.sk-circle .sk-circle7>div{animation-delay:-.6s}.sk-circle .sk-circle8>div{animation-delay:-.5s}.sk-circle .sk-circle9>div{animation-delay:-.4s}.sk-circle .sk-circle10>div{animation-delay:-.3s}.sk-circle .sk-circle11>div{animation-delay:-.2s}.sk-circle .sk-circle12>div{animation-delay:-.1s}@keyframes sk-circleBounceDelay{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"]
                },] }
    ];
    CircleComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var CircleModule = /** @class */ (function () {
        function CircleModule() {
        }
        return CircleModule;
    }());
    CircleModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        CircleComponent,
                    ],
                    exports: [
                        CircleComponent
                    ]
                },] }
    ];

    var CubeGridComponent = /** @class */ (function () {
        function CubeGridComponent() {
            this.color = '#da4733';
        }
        return CubeGridComponent;
    }());
    CubeGridComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-cube-grid',
                    template: "\n    <div class=\"sk-cube-grid\">\n      <div class=\"sk-cube sk-cube1\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube2\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube3\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube4\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube5\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube6\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube7\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube8\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube9\" [style.background-color]=\"color\"></div>\n    </div>",
                    styles: [".sk-cube-grid{height:40px;margin:40px auto;width:40px}.sk-cube-grid .sk-cube{animation:sk-cubeGridScaleDelay 1.3s ease-in-out infinite;float:left;height:33.33%;width:33.33%}.sk-cube-grid .sk-cube1{animation-delay:.2s}.sk-cube-grid .sk-cube2{animation-delay:.3s}.sk-cube-grid .sk-cube3{animation-delay:.4s}.sk-cube-grid .sk-cube4{animation-delay:.1s}.sk-cube-grid .sk-cube5{animation-delay:.2s}.sk-cube-grid .sk-cube6{animation-delay:.3s}.sk-cube-grid .sk-cube7{animation-delay:0s}.sk-cube-grid .sk-cube8{animation-delay:.1s}.sk-cube-grid .sk-cube9{animation-delay:.2s}@keyframes sk-cubeGridScaleDelay{0%,70%,to{transform:scaleX(1)}35%{transform:scale3D(0,0,1)}}"]
                },] }
    ];
    CubeGridComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var CubeGridModule = /** @class */ (function () {
        function CubeGridModule() {
        }
        return CubeGridModule;
    }());
    CubeGridModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        CubeGridComponent,
                    ],
                    exports: [
                        CubeGridComponent
                    ]
                },] }
    ];

    var DoubleBounceComponent = /** @class */ (function () {
        function DoubleBounceComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
        }
        return DoubleBounceComponent;
    }());
    DoubleBounceComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-double-bounce',
                    template: "\n    <div class=\"sk-double-bounce\">\n      <div class=\"sk-child sk-double-bounce1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-child sk-double-bounce2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                    styles: [".sk-double-bounce{height:40px;margin:40px auto;position:relative;width:40px}.sk-double-bounce .sk-child{animation:sk-doubleBounce 2s ease-in-out infinite;border-radius:50%;height:100%;left:0;opacity:.6;position:absolute;top:0;width:100%}.sk-double-bounce .sk-double-bounce2{animation-delay:-1s}@keyframes sk-doubleBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
                },] }
    ];
    DoubleBounceComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var DoubleBounceModule = /** @class */ (function () {
        function DoubleBounceModule() {
        }
        return DoubleBounceModule;
    }());
    DoubleBounceModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        DoubleBounceComponent,
                    ],
                    exports: [
                        DoubleBounceComponent
                    ]
                },] }
    ];

    var FadingCircleComponent = /** @class */ (function () {
        function FadingCircleComponent() {
            this.color = '#da4733';
        }
        return FadingCircleComponent;
    }());
    FadingCircleComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-fading-circle',
                    template: "\n    <div class=\"sk-fading-circle\">\n      <div class=\"sk-circle1 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle2 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle3 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle4 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle5 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle6 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle7 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle8 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle9 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle10 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle11 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle12 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                    styles: [".sk-fading-circle{height:40px;margin:40px auto;position:relative;width:40px}.sk-fading-circle .sk-circle{height:100%;left:0;position:absolute;top:0;width:100%}.sk-fading-circle .sk-circle>div{animation:sk-circleFadeDelay 1.2s ease-in-out infinite both;border-radius:100%;content:\"\";display:block;height:15%;margin:0 auto;width:15%}.sk-fading-circle .sk-circle2{transform:rotate(30deg)}.sk-fading-circle .sk-circle3{transform:rotate(60deg)}.sk-fading-circle .sk-circle4{transform:rotate(90deg)}.sk-fading-circle .sk-circle5{transform:rotate(120deg)}.sk-fading-circle .sk-circle6{transform:rotate(150deg)}.sk-fading-circle .sk-circle7{transform:rotate(180deg)}.sk-fading-circle .sk-circle8{transform:rotate(210deg)}.sk-fading-circle .sk-circle9{transform:rotate(240deg)}.sk-fading-circle .sk-circle10{transform:rotate(270deg)}.sk-fading-circle .sk-circle11{transform:rotate(300deg)}.sk-fading-circle .sk-circle12{transform:rotate(330deg)}.sk-fading-circle .sk-circle2>div{animation-delay:-1.1s}.sk-fading-circle .sk-circle3>div{animation-delay:-1s}.sk-fading-circle .sk-circle4>div{animation-delay:-.9s}.sk-fading-circle .sk-circle5>div{animation-delay:-.8s}.sk-fading-circle .sk-circle6>div{animation-delay:-.7s}.sk-fading-circle .sk-circle7>div{animation-delay:-.6s}.sk-fading-circle .sk-circle8>div{animation-delay:-.5s}.sk-fading-circle .sk-circle9>div{animation-delay:-.4s}.sk-fading-circle .sk-circle10>div{animation-delay:-.3s}.sk-fading-circle .sk-circle11>div{animation-delay:-.2s}.sk-fading-circle .sk-circle12>div{animation-delay:-.1s}@keyframes sk-circleFadeDelay{0%,39%,to{opacity:0}40%{opacity:1}}"]
                },] }
    ];
    FadingCircleComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var FadingCircleModule = /** @class */ (function () {
        function FadingCircleModule() {
        }
        return FadingCircleModule;
    }());
    FadingCircleModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        FadingCircleComponent,
                    ],
                    exports: [
                        FadingCircleComponent
                    ]
                },] }
    ];

    var FoldingCubeComponent = /** @class */ (function () {
        function FoldingCubeComponent() {
            this.color = '#da4733';
        }
        return FoldingCubeComponent;
    }());
    FoldingCubeComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-folding-cube',
                    template: "\n    <div class=\"sk-folding-cube\">\n      <div class=\"sk-cube1 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube2 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube4 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube3 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                    styles: [".sk-folding-cube{height:40px;margin:40px auto;position:relative;transform:rotate(45deg);width:40px}.sk-folding-cube .sk-cube{float:left;height:50%;position:relative;transform:scale(1.1);width:50%}.sk-folding-cube .sk-cube>div{animation:sk-foldCubeAngle 2.4s linear infinite both;content:\"\";height:100%;left:0;position:absolute;top:0;transform-origin:100% 100%;width:100%}.sk-folding-cube .sk-cube2{transform:scale(1.1) rotate(90deg)}.sk-folding-cube .sk-cube3{transform:scale(1.1) rotate(180deg)}.sk-folding-cube .sk-cube4{transform:scale(1.1) rotate(270deg)}.sk-folding-cube .sk-cube2>div{animation-delay:.3s}.sk-folding-cube .sk-cube3>div{animation-delay:.6s}.sk-folding-cube .sk-cube4>div{animation-delay:.9s}@keyframes sk-foldCubeAngle{0%,10%{opacity:0;transform:perspective(140px) rotateX(-180deg)}25%,75%{opacity:1;transform:perspective(140px) rotateX(0deg)}90%,to{opacity:0;transform:perspective(140px) rotateY(180deg)}}"]
                },] }
    ];
    FoldingCubeComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var FoldingCubeModule = /** @class */ (function () {
        function FoldingCubeModule() {
        }
        return FoldingCubeModule;
    }());
    FoldingCubeModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        FoldingCubeComponent,
                    ],
                    exports: [
                        FoldingCubeComponent
                    ]
                },] }
    ];

    var MaterialbarComponent = /** @class */ (function () {
        function MaterialbarComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
            this.colorThree = '#fdba2c';
        }
        return MaterialbarComponent;
    }());
    MaterialbarComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-material-bar',
                    template: "\n  <div class=\"material-bar\">\n    <div class=\"bar\" [style.background-color]=\"colorOne\"></div>\n    <div class=\"bar\" [style.background-color]=\"colorTwo\"></div>\n    <div class=\"bar\" [style.background-color]=\"colorThree\"></div>\n  </div>",
                    styles: [".material-bar{background-color:#fdba2c;height:4px;position:relative;width:100%}.bar{content:\"\";display:inline;height:100%;left:50%;position:absolute;text-align:center;width:0}.bar:first-child{animation:loading 3s linear infinite}.bar:nth-child(2){animation:loading 3s linear 1s infinite}.bar:nth-child(3){animation:loading 3s linear 2s infinite}@keyframes loading{0%{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"]
                },] }
    ];
    MaterialbarComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }],
        colorThree: [{ type: core.Input }]
    };

    var MaterialBarModule = /** @class */ (function () {
        function MaterialBarModule() {
        }
        return MaterialBarModule;
    }());
    MaterialBarModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        MaterialbarComponent,
                    ],
                    exports: [
                        MaterialbarComponent
                    ]
                },] }
    ];

    var PulseComponent = /** @class */ (function () {
        function PulseComponent() {
            this.color = '#da4733';
        }
        return PulseComponent;
    }());
    PulseComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-pulse',
                    template: "<div class=\"sk-spinner sk-spinner-pulse\" [style.background-color]=\"color\"></div>",
                    styles: [".sk-spinner-pulse{animation:sk-pulseScaleOut 1s ease-in-out infinite;background-color:#333;border-radius:100%;height:40px;margin:40px auto;width:40px}@keyframes sk-pulseScaleOut{0%{transform:scale(0)}to{opacity:0;transform:scale(1)}}"]
                },] }
    ];
    PulseComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var PulseModule = /** @class */ (function () {
        function PulseModule() {
        }
        return PulseModule;
    }());
    PulseModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        PulseComponent,
                    ],
                    exports: [
                        PulseComponent
                    ]
                },] }
    ];

    var SlidingBarComponent = /** @class */ (function () {
        function SlidingBarComponent() {
            this.colorOne = '#ddd';
            this.colorTwo = '#3c8dbc';
        }
        return SlidingBarComponent;
    }());
    SlidingBarComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-sliding-bar',
                    template: "<div class=\"sliding-bar\" [style.background-color]=\"colorOne\"><div [style.background-color]=\"colorTwo\"></div></div>",
                    styles: [".sliding-bar{height:4px;overflow:hidden;position:relative;width:100%}.sliding-bar>div{animation:loading 2s linear infinite;background-color:#2980b9;content:\"\";display:block;height:4px;left:-200px;position:absolute;width:200px}@keyframes loading{0%{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"]
                },] }
    ];
    SlidingBarComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var SlidingBarModule = /** @class */ (function () {
        function SlidingBarModule() {
        }
        return SlidingBarModule;
    }());
    SlidingBarModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        SlidingBarComponent,
                    ],
                    exports: [
                        SlidingBarComponent
                    ]
                },] }
    ];

    var ThreeBounceComponent = /** @class */ (function () {
        function ThreeBounceComponent() {
            this.colorOne = '#00a65a';
            this.colorTwo = '#fdba2c';
            this.colorThree = '#da4733';
        }
        return ThreeBounceComponent;
    }());
    ThreeBounceComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-three-bounce',
                    template: "\n    <div class=\"sk-three-bounce\">\n       <div class=\"sk-child sk-bounce1\" [style.background-color]=\"colorOne\"></div>\n       <div class=\"sk-child sk-bounce2\" [style.background-color]=\"colorTwo\"></div>\n       <div class=\"sk-child sk-bounce3\" [style.background-color]=\"colorThree\"></div>\n     </div>",
                    styles: [".sk-three-bounce{margin:40px auto;text-align:center;width:80px}.sk-three-bounce .sk-child{animation:sk-three-bounce 1.4s ease-in-out 0s infinite both;background-color:#333;border-radius:100%;display:inline-block;height:20px;width:20px}.sk-three-bounce .sk-bounce1{animation-delay:-.32s}.sk-three-bounce .sk-bounce2{animation-delay:-.16s}@keyframes sk-three-bounce{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"]
                },] }
    ];
    ThreeBounceComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }],
        colorThree: [{ type: core.Input }]
    };

    var ThreeBounceModule = /** @class */ (function () {
        function ThreeBounceModule() {
        }
        return ThreeBounceModule;
    }());
    ThreeBounceModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        ThreeBounceComponent,
                    ],
                    exports: [
                        ThreeBounceComponent
                    ]
                },] }
    ];

    var WanderingCubesComponent = /** @class */ (function () {
        function WanderingCubesComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
        }
        return WanderingCubesComponent;
    }());
    WanderingCubesComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-wandering-cubes',
                    template: "\n    <div class=\"sk-wandering-cubes\">\n      <div class=\"sk-cube sk-cube1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-cube sk-cube2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                    styles: [".sk-wandering-cubes{height:40px;margin:40px auto;position:relative;width:40px}.sk-wandering-cubes .sk-cube{animation:sk-wanderingCube 1.8s ease-in-out -1.8s infinite both;height:10px;left:0;position:absolute;top:0;width:10px}.sk-wandering-cubes .sk-cube2{animation-delay:-.9s}@keyframes sk-wanderingCube{0%{transform:rotate(0deg)}25%{transform:translateX(30px) rotate(-90deg) scale(.5)}50%{transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}to{transform:rotate(-1turn)}}"]
                },] }
    ];
    WanderingCubesComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var WanderingCubesModule = /** @class */ (function () {
        function WanderingCubesModule() {
        }
        return WanderingCubesModule;
    }());
    WanderingCubesModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        WanderingCubesComponent,
                    ],
                    exports: [
                        WanderingCubesComponent
                    ]
                },] }
    ];

    var WaveComponent = /** @class */ (function () {
        function WaveComponent() {
            this.color = '#da4733';
        }
        return WaveComponent;
    }());
    WaveComponent.decorators = [
        { type: core.Component, args: [{
                    selector: 'mk-wave',
                    template: "\n    <div class=\"sk-wave\">\n      <div class=\"sk-rect sk-rect1\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect2\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect3\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect4\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect5\" [style.background-color]=\"color\"></div>\n    </div>",
                    styles: [".sk-wave{font-size:10px;height:40px;margin:40px auto;text-align:center;width:50px}.sk-wave .sk-rect{animation:sk-waveStretchDelay 1.2s ease-in-out infinite;display:inline-block;height:100%;width:6px}.sk-wave .sk-rect1{animation-delay:-1.2s}.sk-wave .sk-rect2{animation-delay:-1.1s}.sk-wave .sk-rect3{animation-delay:-1s}.sk-wave .sk-rect4{animation-delay:-.9s}.sk-wave .sk-rect5{animation-delay:-.8s}@keyframes sk-waveStretchDelay{0%,40%,to{transform:scaleY(.4)}20%{transform:scaleY(1)}}"]
                },] }
    ];
    WaveComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var WaveModule = /** @class */ (function () {
        function WaveModule() {
        }
        return WaveModule;
    }());
    WaveModule.decorators = [
        { type: core.NgModule, args: [{
                    imports: [
                        common.CommonModule
                    ],
                    declarations: [
                        WaveComponent,
                    ],
                    exports: [
                        WaveComponent
                    ]
                },] }
    ];

    /*
     * Public API Surface of angular-loading-page
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ChasingDotsComponent = ChasingDotsComponent;
    exports.ChasingDotsModule = ChasingDotsModule;
    exports.CircleComponent = CircleComponent;
    exports.CircleModule = CircleModule;
    exports.CubeGridComponent = CubeGridComponent;
    exports.CubeGridModule = CubeGridModule;
    exports.DoubleBounceComponent = DoubleBounceComponent;
    exports.DoubleBounceModule = DoubleBounceModule;
    exports.FadingCircleComponent = FadingCircleComponent;
    exports.FadingCircleModule = FadingCircleModule;
    exports.FoldingCubeComponent = FoldingCubeComponent;
    exports.FoldingCubeModule = FoldingCubeModule;
    exports.HttpInterceptorService = HttpInterceptorService;
    exports.HttpInterceptorServiceFactory = HttpInterceptorServiceFactory;
    exports.HttpObservableService = HttpObservableService;
    exports.LoadingPageDirective = LoadingPageDirective;
    exports.LoadingPageModule = LoadingPageModule;
    exports.MaterialBarModule = MaterialBarModule;
    exports.MaterialbarComponent = MaterialbarComponent;
    exports.PulseComponent = PulseComponent;
    exports.PulseModule = PulseModule;
    exports.RouteInterceptorService = RouteInterceptorService;
    exports.SlidingBarComponent = SlidingBarComponent;
    exports.SlidingBarModule = SlidingBarModule;
    exports.ThreeBounceComponent = ThreeBounceComponent;
    exports.ThreeBounceModule = ThreeBounceModule;
    exports.WanderingCubesComponent = WanderingCubesComponent;
    exports.WanderingCubesModule = WanderingCubesModule;
    exports.WaveComponent = WaveComponent;
    exports.WaveModule = WaveModule;
    exports.ɵ0 = ɵ0;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
