// angular-loading-page/bundles/angular-loading-page.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(require('@angular/core'),require('@angular/router'),require('@angular/common/http'),require('@angular/common'),exports, require('@angular/core'), require('@angular/common'), require('@angular/common/http'), require('@angular/router'), require('rxjs'), require('rxjs/operators')) :
    typeof define === 'function' && define.amd ? define('angular-loading-page', ['@angular/core','@angular/router','@angular/common/http','@angular/common','exports', '@angular/core', '@angular/common', '@angular/common/http', '@angular/router', 'rxjs', 'rxjs/operators'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.ng.core,global.ng.router,global.ng.common.http,global.ng.common,global['angular-loading-page'] = {}, global.ng.core, global.ng.common, global.ng.common.http, global.ng.router, global.rxjs, global.rxjs.operators));
}(this, (function (ɵngcc0,ɵngcc1,ɵngcc2,ɵngcc3,exports, core, common, http, router, rxjs, operators) { 'use strict';

    var RouteInterceptorService = /** @class */ (function () {
        /**
         * @method constructor
         */
        function RouteInterceptorService(router) {
            var _this = this;
            this.router = router;
            this.isPendingRouteSubject = new rxjs.Subject();
            if (this.router) {
                router.events.subscribe(function (event) {
                    _this.routerInterceptor(event);
                });
            }
        }
        Object.defineProperty(RouteInterceptorService.prototype, "isPendingRoute", {
            /**
             * [isPendingRoute description]
             * @method isPendingRoute
             * @return [description]
             */
            get: function () {
                return this.isPendingRouteSubject.asObservable();
            },
            enumerable: false,
            configurable: true
        });
        /**
         * [routerInterceptor description]
         * @method routerInterceptor
         * @param event [description]
         */
        RouteInterceptorService.prototype.routerInterceptor = function (event) {
            if (event instanceof router.NavigationStart) {
                this.onRouteStart();
            }
            if (event instanceof router.NavigationEnd || event instanceof router.NavigationCancel || event instanceof router.NavigationError) {
                this.onRouteDone();
            }
        };
        /**
         * [onRequestStart description]
         * @method onRequestStart
         */
        RouteInterceptorService.prototype.onRouteStart = function () {
            this.isPendingRouteSubject.next(true);
        };
        /**
         * [onRequestDone description]
         * @method onRequestDone
         */
        RouteInterceptorService.prototype.onRouteDone = function () {
            this.isPendingRouteSubject.next(false);
        };
RouteInterceptorService.ɵfac = function RouteInterceptorService_Factory(t) { return new (t || RouteInterceptorService)(ɵngcc0.ɵɵinject(ɵngcc1.Router, 8)); };
RouteInterceptorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: RouteInterceptorService, factory: function (t) { return RouteInterceptorService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(RouteInterceptorService, [{
        type: core.Injectable
    }], function () { return [{ type: ɵngcc1.Router, decorators: [{
                type: core.Optional
            }] }]; }, null); })();
        return RouteInterceptorService;
    }());
    RouteInterceptorService.ctorParameters = function () { return [
        { type: router.Router, decorators: [{ type: core.Optional }] }
    ]; };

    var HttpObservableService = /** @class */ (function () {
        function HttpObservableService() {
            this.pendingRequestsCounter = 0;
            this.isPendingRequestsSubject = new rxjs.Subject();
        }
        Object.defineProperty(HttpObservableService.prototype, "isPendingRequests", {
            /**
             * [isPendingRequests description]
             * @method isPendingRequests
             * @return [description]
             */
            get: function () {
                return this.isPendingRequestsSubject.asObservable();
            },
            enumerable: false,
            configurable: true
        });
        Object.defineProperty(HttpObservableService.prototype, "pendingRequests", {
            /**
             * [pendingRequests description]
             * @method pendingRequests
             * @return [description]
             */
            get: function () {
                return this.pendingRequestsCounter;
            },
            enumerable: false,
            configurable: true
        });
        /**
         * [onRequestStart description]
         * @method onRequestStart
         */
        HttpObservableService.prototype.onRequestStart = function () {
            this.pendingRequestsCounter++;
            if (this.pendingRequestsCounter === 1) {
                this.isPendingRequestsSubject.next(true);
            }
        };
        /**
         * [onRequestDone description]
         * @method onRequestDone
         */
        HttpObservableService.prototype.onRequestDone = function () {
            this.pendingRequestsCounter--;
            if (this.pendingRequestsCounter === 0) {
                this.isPendingRequestsSubject.next(false);
            }
        };
HttpObservableService.ɵfac = function HttpObservableService_Factory(t) { return new (t || HttpObservableService)(); };
HttpObservableService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: HttpObservableService, factory: function (t) { return HttpObservableService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HttpObservableService, [{
        type: core.Injectable
    }], function () { return []; }, null); })();
        return HttpObservableService;
    }());

    var HttpInterceptorService = /** @class */ (function () {
        /**
         * @method constructor
         * @param injector [description]
         */
        function HttpInterceptorService(injector) {
            this.injector = injector;
            this.httpObservableService = injector.get(HttpObservableService);
        }
        /**
         * @method intercept
         * @param request [description]
         * @param next [description]
         */
        HttpInterceptorService.prototype.intercept = function (request, next) {
            var _this = this;
            this.httpObservableService.onRequestStart();
            return next.handle(request).pipe(operators.finalize(function () {
                _this.httpObservableService.onRequestDone();
            }));
        };
HttpInterceptorService.ɵfac = function HttpInterceptorService_Factory(t) { return new (t || HttpInterceptorService)(ɵngcc0.ɵɵinject(ɵngcc0.Injector)); };
HttpInterceptorService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: HttpInterceptorService, factory: function (t) { return HttpInterceptorService.ɵfac(t); } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HttpInterceptorService, [{
        type: core.Injectable
    }], function () { return [{ type: ɵngcc0.Injector }]; }, null); })();
        return HttpInterceptorService;
    }());
    HttpInterceptorService.ctorParameters = function () { return [
        { type: core.Injector }
    ]; };
    function HttpInterceptorServiceFactory(injector) {
        return new HttpInterceptorService(injector);
    }

    /*
     * mkLoadingPage Directive
     */
    var LoadingPageDirective = /** @class */ (function () {
        /**
         * @method constructor
         * @param privatehttpObservableService   [description]
         * @param privaterouteInterceptorService [description]
         * @param privatetemplateRef             [description]
         * @param privateviewContainer           [description]                                              [description]
         */
        function LoadingPageDirective(httpObservableService, routeInterceptorService, templateRef, viewContainer) {
            this.httpObservableService = httpObservableService;
            this.routeInterceptorService = routeInterceptorService;
            this.templateRef = templateRef;
            this.viewContainer = viewContainer;
            this.isPendingRequests = false;
            this.isPendingRoute = false;
            this.hasView = false;
        }
        /**
         * @method ngOnChanges
         * @param changes [description]
         */
        LoadingPageDirective.prototype.ngOnChanges = function (changes) {
            if (changes.mkLoadingPage.currentValue) {
                this.mkLoadingPage = changes.mkLoadingPage.currentValue;
            }
            if (changes.mkLoadingPage.firstChange) {
                this.init();
            }
        };
        /**
         * [init description]
         * @method init
         * @return [description]
         */
        LoadingPageDirective.prototype.init = function () {
            var _this = this;
            if (!this.mkLoadingPage || this.mkLoadingPage.checkPendingHttp) {
                this.httpObservableService.isPendingRequests.subscribe(function (value) {
                    _this.isPendingRequests = value;
                    _this.isVisible();
                });
            }
            if (!this.mkLoadingPage || this.mkLoadingPage.checkPendingRoute) {
                this.routeInterceptorService.isPendingRoute.subscribe(function (value) {
                    _this.isPendingRoute = value;
                    _this.isVisible();
                });
            }
        };
        /**
         * [isVisible description]
         * @method isVisible
         * @return [description]
         */
        LoadingPageDirective.prototype.isVisible = function () {
            if (this.hasView && !this.isPendingRequests && !this.isPendingRoute) {
                this.viewContainer.clear();
                this.hasView = false;
            }
            else if (!this.hasView && (this.isPendingRequests || this.isPendingRoute)) {
                this.viewContainer.createEmbeddedView(this.templateRef);
                this.hasView = true;
            }
        };
LoadingPageDirective.ɵfac = function LoadingPageDirective_Factory(t) { return new (t || LoadingPageDirective)(ɵngcc0.ɵɵdirectiveInject(HttpObservableService), ɵngcc0.ɵɵdirectiveInject(RouteInterceptorService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.TemplateRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef)); };
LoadingPageDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: LoadingPageDirective, selectors: [["", "mkLoadingPage", ""]], inputs: { mkLoadingPage: "mkLoadingPage" }, features: [ɵngcc0.ɵɵNgOnChangesFeature] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoadingPageDirective, [{
        type: core.Directive,
        args: [{
                selector: '[mkLoadingPage]'
            }]
    }], function () { return [{ type: HttpObservableService }, { type: RouteInterceptorService }, { type: ɵngcc0.TemplateRef }, { type: ɵngcc0.ViewContainerRef }]; }, { mkLoadingPage: [{
            type: core.Input
        }] }); })();
        return LoadingPageDirective;
    }());
    LoadingPageDirective.ctorParameters = function () { return [
        { type: HttpObservableService },
        { type: RouteInterceptorService },
        { type: core.TemplateRef },
        { type: core.ViewContainerRef }
    ]; };
    LoadingPageDirective.propDecorators = {
        mkLoadingPage: [{ type: core.Input }]
    };

    var HttpInterceptorServiceExistingProvider = {
        provide: http.HTTP_INTERCEPTORS,
        useExisting: HttpInterceptorService,
        multi: true
    };
    var ɵ0 = HttpInterceptorServiceFactory;
    var LoadingPageModule = /** @class */ (function () {
        function LoadingPageModule() {
        }
LoadingPageModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LoadingPageModule });
LoadingPageModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function LoadingPageModule_Factory(t) { return new (t || LoadingPageModule)(); }, providers: [
        RouteInterceptorService,
        HttpObservableService,
        HttpInterceptorServiceExistingProvider,
        {
            provide: HttpInterceptorService,
            useFactory: ɵ0,
            deps: [core.Injector]
        }
    ], imports: [[
            common.CommonModule,
            http.HttpClientModule
        ], ɵngcc2.HttpClientModule] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LoadingPageModule, { declarations: [LoadingPageDirective], imports: [ɵngcc3.CommonModule, ɵngcc2.HttpClientModule], exports: [ɵngcc2.HttpClientModule, LoadingPageDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LoadingPageModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule,
                    http.HttpClientModule
                ],
                declarations: [
                    LoadingPageDirective
                ],
                exports: [
                    http.HttpClientModule,
                    LoadingPageDirective
                ],
                providers: [
                    RouteInterceptorService,
                    HttpObservableService,
                    HttpInterceptorServiceExistingProvider,
                    {
                        provide: HttpInterceptorService,
                        useFactory: ɵ0,
                        deps: [core.Injector]
                    }
                ]
            }]
    }], function () { return []; }, null); })();
        return LoadingPageModule;
    }());

    var ChasingDotsComponent = /** @class */ (function () {
        function ChasingDotsComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
        }
ChasingDotsComponent.ɵfac = function ChasingDotsComponent_Factory(t) { return new (t || ChasingDotsComponent)(); };
ChasingDotsComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ChasingDotsComponent, selectors: [["mk-chasing-dots"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-chasing-dots"], [1, "sk-child", "sk-dot1"], [1, "sk-child", "sk-dot2"]], template: function ChasingDotsComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-chasing-dots[_ngcontent-%COMP%]{animation:sk-chasingDotsRotate 2s linear infinite;height:40px;margin:40px auto;position:relative;text-align:center;width:40px}.sk-chasing-dots[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{animation:sk-chasingDotsBounce 2s ease-in-out infinite;border-radius:100%;display:inline-block;height:60%;position:absolute;top:0;width:60%}.sk-chasing-dots[_ngcontent-%COMP%]   .sk-dot2[_ngcontent-%COMP%]{animation-delay:-1s;bottom:0;top:auto}@keyframes sk-chasingDotsRotate{to{transform:rotate(1turn)}}@keyframes sk-chasingDotsBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChasingDotsComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-chasing-dots',
                template: "\n    <div class=\"sk-chasing-dots\">\n      <div class=\"sk-child sk-dot1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-child sk-dot2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                styles: [".sk-chasing-dots{animation:sk-chasingDotsRotate 2s linear infinite;height:40px;margin:40px auto;position:relative;text-align:center;width:40px}.sk-chasing-dots .sk-child{animation:sk-chasingDotsBounce 2s ease-in-out infinite;border-radius:100%;display:inline-block;height:60%;position:absolute;top:0;width:60%}.sk-chasing-dots .sk-dot2{animation-delay:-1s;bottom:0;top:auto}@keyframes sk-chasingDotsRotate{to{transform:rotate(1turn)}}@keyframes sk-chasingDotsBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: core.Input
        }], colorTwo: [{
            type: core.Input
        }] }); })();
        return ChasingDotsComponent;
    }());
    ChasingDotsComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var ChasingDotsModule = /** @class */ (function () {
        function ChasingDotsModule() {
        }
ChasingDotsModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChasingDotsModule });
ChasingDotsModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChasingDotsModule_Factory(t) { return new (t || ChasingDotsModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChasingDotsModule, { declarations: [ChasingDotsComponent], imports: [ɵngcc3.CommonModule], exports: [ChasingDotsComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChasingDotsModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    ChasingDotsComponent,
                ],
                exports: [
                    ChasingDotsComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return ChasingDotsModule;
    }());

    var CircleComponent = /** @class */ (function () {
        function CircleComponent() {
            this.color = '#da4733';
        }
CircleComponent.ɵfac = function CircleComponent_Factory(t) { return new (t || CircleComponent)(); };
CircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CircleComponent, selectors: [["mk-circle"]], inputs: { color: "color" }, decls: 25, vars: 24, consts: [[1, "sk-circle"], [1, "sk-circle1", "sk-child"], [1, "sk-circle2", "sk-child"], [1, "sk-circle3", "sk-child"], [1, "sk-circle4", "sk-child"], [1, "sk-circle5", "sk-child"], [1, "sk-circle6", "sk-child"], [1, "sk-circle7", "sk-child"], [1, "sk-circle8", "sk-child"], [1, "sk-circle9", "sk-child"], [1, "sk-circle10", "sk-child"], [1, "sk-circle11", "sk-child"], [1, "sk-circle12", "sk-child"]], template: function CircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 5);
        ɵngcc0.ɵɵelement(10, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 6);
        ɵngcc0.ɵɵelement(12, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 7);
        ɵngcc0.ɵɵelement(14, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "div", 8);
        ɵngcc0.ɵɵelement(16, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div", 9);
        ɵngcc0.ɵɵelement(18, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 10);
        ɵngcc0.ɵɵelement(20, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 11);
        ɵngcc0.ɵɵelement(22, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(23, "div", 12);
        ɵngcc0.ɵɵelement(24, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-circle[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-circle[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{height:100%;left:0;position:absolute;top:0;width:100%}.sk-circle[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:sk-circleBounceDelay 1.2s ease-in-out infinite both;border-radius:100%;display:block;height:15%;margin:0 auto;width:15%}.sk-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%]{transform:rotate(30deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%]{transform:rotate(60deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%]{transform:rotate(90deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%]{transform:rotate(120deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%]{transform:rotate(150deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%]{transform:rotate(180deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%]{transform:rotate(210deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%]{transform:rotate(240deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%]{transform:rotate(270deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%]{transform:rotate(300deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%]{transform:rotate(330deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1.1s}.sk-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1s}.sk-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.9s}.sk-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.8s}.sk-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.7s}.sk-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.6s}.sk-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.5s}.sk-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.4s}.sk-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.3s}.sk-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.2s}.sk-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.1s}@keyframes sk-circleBounceDelay{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-circle',
                template: "\n    <div class=\"sk-circle\">\n      <div class=\"sk-circle1 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle2 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle3 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle4 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle5 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle6 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle7 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle8 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle9 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle10 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle11 sk-child\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle12 sk-child\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                styles: [".sk-circle{height:40px;margin:40px auto;position:relative;width:40px}.sk-circle .sk-child{height:100%;left:0;position:absolute;top:0;width:100%}.sk-circle .sk-child>div{animation:sk-circleBounceDelay 1.2s ease-in-out infinite both;border-radius:100%;display:block;height:15%;margin:0 auto;width:15%}.sk-circle .sk-circle2{transform:rotate(30deg)}.sk-circle .sk-circle3{transform:rotate(60deg)}.sk-circle .sk-circle4{transform:rotate(90deg)}.sk-circle .sk-circle5{transform:rotate(120deg)}.sk-circle .sk-circle6{transform:rotate(150deg)}.sk-circle .sk-circle7{transform:rotate(180deg)}.sk-circle .sk-circle8{transform:rotate(210deg)}.sk-circle .sk-circle9{transform:rotate(240deg)}.sk-circle .sk-circle10{transform:rotate(270deg)}.sk-circle .sk-circle11{transform:rotate(300deg)}.sk-circle .sk-circle12{transform:rotate(330deg)}.sk-circle .sk-circle2>div{animation-delay:-1.1s}.sk-circle .sk-circle3>div{animation-delay:-1s}.sk-circle .sk-circle4>div{animation-delay:-.9s}.sk-circle .sk-circle5>div{animation-delay:-.8s}.sk-circle .sk-circle6>div{animation-delay:-.7s}.sk-circle .sk-circle7>div{animation-delay:-.6s}.sk-circle .sk-circle8>div{animation-delay:-.5s}.sk-circle .sk-circle9>div{animation-delay:-.4s}.sk-circle .sk-circle10>div{animation-delay:-.3s}.sk-circle .sk-circle11>div{animation-delay:-.2s}.sk-circle .sk-circle12>div{animation-delay:-.1s}@keyframes sk-circleBounceDelay{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: core.Input
        }] }); })();
        return CircleComponent;
    }());
    CircleComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var CircleModule = /** @class */ (function () {
        function CircleModule() {
        }
CircleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CircleModule });
CircleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CircleModule_Factory(t) { return new (t || CircleModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CircleModule, { declarations: [CircleComponent], imports: [ɵngcc3.CommonModule], exports: [CircleComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    CircleComponent,
                ],
                exports: [
                    CircleComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return CircleModule;
    }());

    var CubeGridComponent = /** @class */ (function () {
        function CubeGridComponent() {
            this.color = '#da4733';
        }
CubeGridComponent.ɵfac = function CubeGridComponent_Factory(t) { return new (t || CubeGridComponent)(); };
CubeGridComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CubeGridComponent, selectors: [["mk-cube-grid"]], inputs: { color: "color" }, decls: 10, vars: 18, consts: [[1, "sk-cube-grid"], [1, "sk-cube", "sk-cube1"], [1, "sk-cube", "sk-cube2"], [1, "sk-cube", "sk-cube3"], [1, "sk-cube", "sk-cube4"], [1, "sk-cube", "sk-cube5"], [1, "sk-cube", "sk-cube6"], [1, "sk-cube", "sk-cube7"], [1, "sk-cube", "sk-cube8"], [1, "sk-cube", "sk-cube9"]], template: function CubeGridComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelement(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelement(6, "div", 6);
        ɵngcc0.ɵɵelement(7, "div", 7);
        ɵngcc0.ɵɵelement(8, "div", 8);
        ɵngcc0.ɵɵelement(9, "div", 9);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-cube-grid[_ngcontent-%COMP%]{height:40px;margin:40px auto;width:40px}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{animation:sk-cubeGridScaleDelay 1.3s ease-in-out infinite;float:left;height:33.33%;width:33.33%}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube1[_ngcontent-%COMP%]{animation-delay:.2s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{animation-delay:.3s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%]{animation-delay:.4s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%]{animation-delay:.1s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube5[_ngcontent-%COMP%]{animation-delay:.2s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube6[_ngcontent-%COMP%]{animation-delay:.3s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube7[_ngcontent-%COMP%]{animation-delay:0s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube8[_ngcontent-%COMP%]{animation-delay:.1s}.sk-cube-grid[_ngcontent-%COMP%]   .sk-cube9[_ngcontent-%COMP%]{animation-delay:.2s}@keyframes sk-cubeGridScaleDelay{0%,70%,to{transform:scaleX(1)}35%{transform:scale3D(0,0,1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CubeGridComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-cube-grid',
                template: "\n    <div class=\"sk-cube-grid\">\n      <div class=\"sk-cube sk-cube1\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube2\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube3\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube4\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube5\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube6\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube7\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube8\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-cube sk-cube9\" [style.background-color]=\"color\"></div>\n    </div>",
                styles: [".sk-cube-grid{height:40px;margin:40px auto;width:40px}.sk-cube-grid .sk-cube{animation:sk-cubeGridScaleDelay 1.3s ease-in-out infinite;float:left;height:33.33%;width:33.33%}.sk-cube-grid .sk-cube1{animation-delay:.2s}.sk-cube-grid .sk-cube2{animation-delay:.3s}.sk-cube-grid .sk-cube3{animation-delay:.4s}.sk-cube-grid .sk-cube4{animation-delay:.1s}.sk-cube-grid .sk-cube5{animation-delay:.2s}.sk-cube-grid .sk-cube6{animation-delay:.3s}.sk-cube-grid .sk-cube7{animation-delay:0s}.sk-cube-grid .sk-cube8{animation-delay:.1s}.sk-cube-grid .sk-cube9{animation-delay:.2s}@keyframes sk-cubeGridScaleDelay{0%,70%,to{transform:scaleX(1)}35%{transform:scale3D(0,0,1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: core.Input
        }] }); })();
        return CubeGridComponent;
    }());
    CubeGridComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var CubeGridModule = /** @class */ (function () {
        function CubeGridModule() {
        }
CubeGridModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CubeGridModule });
CubeGridModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CubeGridModule_Factory(t) { return new (t || CubeGridModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CubeGridModule, { declarations: [CubeGridComponent], imports: [ɵngcc3.CommonModule], exports: [CubeGridComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CubeGridModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    CubeGridComponent,
                ],
                exports: [
                    CubeGridComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return CubeGridModule;
    }());

    var DoubleBounceComponent = /** @class */ (function () {
        function DoubleBounceComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
        }
DoubleBounceComponent.ɵfac = function DoubleBounceComponent_Factory(t) { return new (t || DoubleBounceComponent)(); };
DoubleBounceComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DoubleBounceComponent, selectors: [["mk-double-bounce"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-double-bounce"], [1, "sk-child", "sk-double-bounce1"], [1, "sk-child", "sk-double-bounce2"]], template: function DoubleBounceComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-double-bounce[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-double-bounce[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{animation:sk-doubleBounce 2s ease-in-out infinite;border-radius:50%;height:100%;left:0;opacity:.6;position:absolute;top:0;width:100%}.sk-double-bounce[_ngcontent-%COMP%]   .sk-double-bounce2[_ngcontent-%COMP%]{animation-delay:-1s}@keyframes sk-doubleBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DoubleBounceComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-double-bounce',
                template: "\n    <div class=\"sk-double-bounce\">\n      <div class=\"sk-child sk-double-bounce1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-child sk-double-bounce2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                styles: [".sk-double-bounce{height:40px;margin:40px auto;position:relative;width:40px}.sk-double-bounce .sk-child{animation:sk-doubleBounce 2s ease-in-out infinite;border-radius:50%;height:100%;left:0;opacity:.6;position:absolute;top:0;width:100%}.sk-double-bounce .sk-double-bounce2{animation-delay:-1s}@keyframes sk-doubleBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: core.Input
        }], colorTwo: [{
            type: core.Input
        }] }); })();
        return DoubleBounceComponent;
    }());
    DoubleBounceComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var DoubleBounceModule = /** @class */ (function () {
        function DoubleBounceModule() {
        }
DoubleBounceModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DoubleBounceModule });
DoubleBounceModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DoubleBounceModule_Factory(t) { return new (t || DoubleBounceModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DoubleBounceModule, { declarations: [DoubleBounceComponent], imports: [ɵngcc3.CommonModule], exports: [DoubleBounceComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DoubleBounceModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    DoubleBounceComponent,
                ],
                exports: [
                    DoubleBounceComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return DoubleBounceModule;
    }());

    var FadingCircleComponent = /** @class */ (function () {
        function FadingCircleComponent() {
            this.color = '#da4733';
        }
FadingCircleComponent.ɵfac = function FadingCircleComponent_Factory(t) { return new (t || FadingCircleComponent)(); };
FadingCircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FadingCircleComponent, selectors: [["mk-fading-circle"]], inputs: { color: "color" }, decls: 25, vars: 24, consts: [[1, "sk-fading-circle"], [1, "sk-circle1", "sk-circle"], [1, "sk-circle2", "sk-circle"], [1, "sk-circle3", "sk-circle"], [1, "sk-circle4", "sk-circle"], [1, "sk-circle5", "sk-circle"], [1, "sk-circle6", "sk-circle"], [1, "sk-circle7", "sk-circle"], [1, "sk-circle8", "sk-circle"], [1, "sk-circle9", "sk-circle"], [1, "sk-circle10", "sk-circle"], [1, "sk-circle11", "sk-circle"], [1, "sk-circle12", "sk-circle"]], template: function FadingCircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 5);
        ɵngcc0.ɵɵelement(10, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 6);
        ɵngcc0.ɵɵelement(12, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 7);
        ɵngcc0.ɵɵelement(14, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "div", 8);
        ɵngcc0.ɵɵelement(16, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div", 9);
        ɵngcc0.ɵɵelement(18, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 10);
        ɵngcc0.ɵɵelement(20, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 11);
        ɵngcc0.ɵɵelement(22, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(23, "div", 12);
        ɵngcc0.ɵɵelement(24, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-fading-circle[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle[_ngcontent-%COMP%]{height:100%;left:0;position:absolute;top:0;width:100%}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:sk-circleFadeDelay 1.2s ease-in-out infinite both;border-radius:100%;content:\"\";display:block;height:15%;margin:0 auto;width:15%}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%]{transform:rotate(30deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%]{transform:rotate(60deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%]{transform:rotate(90deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%]{transform:rotate(120deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%]{transform:rotate(150deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%]{transform:rotate(180deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%]{transform:rotate(210deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%]{transform:rotate(240deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%]{transform:rotate(270deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%]{transform:rotate(300deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%]{transform:rotate(330deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1.1s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.9s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.8s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.7s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.6s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.5s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.4s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.3s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.2s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.1s}@keyframes sk-circleFadeDelay{0%,39%,to{opacity:0}40%{opacity:1}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FadingCircleComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-fading-circle',
                template: "\n    <div class=\"sk-fading-circle\">\n      <div class=\"sk-circle1 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle2 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle3 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle4 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle5 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle6 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle7 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle8 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle9 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle10 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle11 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-circle12 sk-circle\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                styles: [".sk-fading-circle{height:40px;margin:40px auto;position:relative;width:40px}.sk-fading-circle .sk-circle{height:100%;left:0;position:absolute;top:0;width:100%}.sk-fading-circle .sk-circle>div{animation:sk-circleFadeDelay 1.2s ease-in-out infinite both;border-radius:100%;content:\"\";display:block;height:15%;margin:0 auto;width:15%}.sk-fading-circle .sk-circle2{transform:rotate(30deg)}.sk-fading-circle .sk-circle3{transform:rotate(60deg)}.sk-fading-circle .sk-circle4{transform:rotate(90deg)}.sk-fading-circle .sk-circle5{transform:rotate(120deg)}.sk-fading-circle .sk-circle6{transform:rotate(150deg)}.sk-fading-circle .sk-circle7{transform:rotate(180deg)}.sk-fading-circle .sk-circle8{transform:rotate(210deg)}.sk-fading-circle .sk-circle9{transform:rotate(240deg)}.sk-fading-circle .sk-circle10{transform:rotate(270deg)}.sk-fading-circle .sk-circle11{transform:rotate(300deg)}.sk-fading-circle .sk-circle12{transform:rotate(330deg)}.sk-fading-circle .sk-circle2>div{animation-delay:-1.1s}.sk-fading-circle .sk-circle3>div{animation-delay:-1s}.sk-fading-circle .sk-circle4>div{animation-delay:-.9s}.sk-fading-circle .sk-circle5>div{animation-delay:-.8s}.sk-fading-circle .sk-circle6>div{animation-delay:-.7s}.sk-fading-circle .sk-circle7>div{animation-delay:-.6s}.sk-fading-circle .sk-circle8>div{animation-delay:-.5s}.sk-fading-circle .sk-circle9>div{animation-delay:-.4s}.sk-fading-circle .sk-circle10>div{animation-delay:-.3s}.sk-fading-circle .sk-circle11>div{animation-delay:-.2s}.sk-fading-circle .sk-circle12>div{animation-delay:-.1s}@keyframes sk-circleFadeDelay{0%,39%,to{opacity:0}40%{opacity:1}}"]
            }]
    }], function () { return []; }, { color: [{
            type: core.Input
        }] }); })();
        return FadingCircleComponent;
    }());
    FadingCircleComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var FadingCircleModule = /** @class */ (function () {
        function FadingCircleModule() {
        }
FadingCircleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FadingCircleModule });
FadingCircleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FadingCircleModule_Factory(t) { return new (t || FadingCircleModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FadingCircleModule, { declarations: [FadingCircleComponent], imports: [ɵngcc3.CommonModule], exports: [FadingCircleComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FadingCircleModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    FadingCircleComponent,
                ],
                exports: [
                    FadingCircleComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return FadingCircleModule;
    }());

    var FoldingCubeComponent = /** @class */ (function () {
        function FoldingCubeComponent() {
            this.color = '#da4733';
        }
FoldingCubeComponent.ɵfac = function FoldingCubeComponent_Factory(t) { return new (t || FoldingCubeComponent)(); };
FoldingCubeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FoldingCubeComponent, selectors: [["mk-folding-cube"]], inputs: { color: "color" }, decls: 9, vars: 8, consts: [[1, "sk-folding-cube"], [1, "sk-cube1", "sk-cube"], [1, "sk-cube2", "sk-cube"], [1, "sk-cube4", "sk-cube"], [1, "sk-cube3", "sk-cube"]], template: function FoldingCubeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-folding-cube[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;transform:rotate(45deg);width:40px}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{float:left;height:50%;position:relative;transform:scale(1.1);width:50%}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:sk-foldCubeAngle 2.4s linear infinite both;content:\"\";height:100%;left:0;position:absolute;top:0;transform-origin:100% 100%;width:100%}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{transform:scale(1.1) rotate(90deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%]{transform:scale(1.1) rotate(180deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%]{transform:scale(1.1) rotate(270deg)}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:.3s}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:.6s}.sk-folding-cube[_ngcontent-%COMP%]   .sk-cube4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:.9s}@keyframes sk-foldCubeAngle{0%,10%{opacity:0;transform:perspective(140px) rotateX(-180deg)}25%,75%{opacity:1;transform:perspective(140px) rotateX(0deg)}90%,to{opacity:0;transform:perspective(140px) rotateY(180deg)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FoldingCubeComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-folding-cube',
                template: "\n    <div class=\"sk-folding-cube\">\n      <div class=\"sk-cube1 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube2 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube4 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n      <div class=\"sk-cube3 sk-cube\"><div [style.background-color]=\"color\"></div></div>\n    </div>",
                styles: [".sk-folding-cube{height:40px;margin:40px auto;position:relative;transform:rotate(45deg);width:40px}.sk-folding-cube .sk-cube{float:left;height:50%;position:relative;transform:scale(1.1);width:50%}.sk-folding-cube .sk-cube>div{animation:sk-foldCubeAngle 2.4s linear infinite both;content:\"\";height:100%;left:0;position:absolute;top:0;transform-origin:100% 100%;width:100%}.sk-folding-cube .sk-cube2{transform:scale(1.1) rotate(90deg)}.sk-folding-cube .sk-cube3{transform:scale(1.1) rotate(180deg)}.sk-folding-cube .sk-cube4{transform:scale(1.1) rotate(270deg)}.sk-folding-cube .sk-cube2>div{animation-delay:.3s}.sk-folding-cube .sk-cube3>div{animation-delay:.6s}.sk-folding-cube .sk-cube4>div{animation-delay:.9s}@keyframes sk-foldCubeAngle{0%,10%{opacity:0;transform:perspective(140px) rotateX(-180deg)}25%,75%{opacity:1;transform:perspective(140px) rotateX(0deg)}90%,to{opacity:0;transform:perspective(140px) rotateY(180deg)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: core.Input
        }] }); })();
        return FoldingCubeComponent;
    }());
    FoldingCubeComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var FoldingCubeModule = /** @class */ (function () {
        function FoldingCubeModule() {
        }
FoldingCubeModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FoldingCubeModule });
FoldingCubeModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FoldingCubeModule_Factory(t) { return new (t || FoldingCubeModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FoldingCubeModule, { declarations: [FoldingCubeComponent], imports: [ɵngcc3.CommonModule], exports: [FoldingCubeComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FoldingCubeModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    FoldingCubeComponent,
                ],
                exports: [
                    FoldingCubeComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return FoldingCubeModule;
    }());

    var MaterialbarComponent = /** @class */ (function () {
        function MaterialbarComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
            this.colorThree = '#fdba2c';
        }
MaterialbarComponent.ɵfac = function MaterialbarComponent_Factory(t) { return new (t || MaterialbarComponent)(); };
MaterialbarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MaterialbarComponent, selectors: [["mk-material-bar"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo", colorThree: "colorThree" }, decls: 4, vars: 6, consts: [[1, "material-bar"], [1, "bar"]], template: function MaterialbarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 1);
        ɵngcc0.ɵɵelement(3, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorThree);
    } }, styles: [".material-bar[_ngcontent-%COMP%]{background-color:#fdba2c;height:4px;position:relative;width:100%}.bar[_ngcontent-%COMP%]{content:\"\";display:inline;height:100%;left:50%;position:absolute;text-align:center;width:0}.bar[_ngcontent-%COMP%]:first-child{animation:loading 3s linear infinite}.bar[_ngcontent-%COMP%]:nth-child(2){animation:loading 3s linear 1s infinite}.bar[_ngcontent-%COMP%]:nth-child(3){animation:loading 3s linear 2s infinite}@keyframes loading{0%{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialbarComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-material-bar',
                template: "\n  <div class=\"material-bar\">\n    <div class=\"bar\" [style.background-color]=\"colorOne\"></div>\n    <div class=\"bar\" [style.background-color]=\"colorTwo\"></div>\n    <div class=\"bar\" [style.background-color]=\"colorThree\"></div>\n  </div>",
                styles: [".material-bar{background-color:#fdba2c;height:4px;position:relative;width:100%}.bar{content:\"\";display:inline;height:100%;left:50%;position:absolute;text-align:center;width:0}.bar:first-child{animation:loading 3s linear infinite}.bar:nth-child(2){animation:loading 3s linear 1s infinite}.bar:nth-child(3){animation:loading 3s linear 2s infinite}@keyframes loading{0%{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: core.Input
        }], colorTwo: [{
            type: core.Input
        }], colorThree: [{
            type: core.Input
        }] }); })();
        return MaterialbarComponent;
    }());
    MaterialbarComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }],
        colorThree: [{ type: core.Input }]
    };

    var MaterialBarModule = /** @class */ (function () {
        function MaterialBarModule() {
        }
MaterialBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MaterialBarModule });
MaterialBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MaterialBarModule_Factory(t) { return new (t || MaterialBarModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MaterialBarModule, { declarations: [MaterialbarComponent], imports: [ɵngcc3.CommonModule], exports: [MaterialbarComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialBarModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    MaterialbarComponent,
                ],
                exports: [
                    MaterialbarComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return MaterialBarModule;
    }());

    var PulseComponent = /** @class */ (function () {
        function PulseComponent() {
            this.color = '#da4733';
        }
PulseComponent.ɵfac = function PulseComponent_Factory(t) { return new (t || PulseComponent)(); };
PulseComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PulseComponent, selectors: [["mk-pulse"]], inputs: { color: "color" }, decls: 1, vars: 2, consts: [[1, "sk-spinner", "sk-spinner-pulse"]], template: function PulseComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelement(0, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-spinner-pulse[_ngcontent-%COMP%]{animation:sk-pulseScaleOut 1s ease-in-out infinite;background-color:#333;border-radius:100%;height:40px;margin:40px auto;width:40px}@keyframes sk-pulseScaleOut{0%{transform:scale(0)}to{opacity:0;transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PulseComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-pulse',
                template: "<div class=\"sk-spinner sk-spinner-pulse\" [style.background-color]=\"color\"></div>",
                styles: [".sk-spinner-pulse{animation:sk-pulseScaleOut 1s ease-in-out infinite;background-color:#333;border-radius:100%;height:40px;margin:40px auto;width:40px}@keyframes sk-pulseScaleOut{0%{transform:scale(0)}to{opacity:0;transform:scale(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: core.Input
        }] }); })();
        return PulseComponent;
    }());
    PulseComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var PulseModule = /** @class */ (function () {
        function PulseModule() {
        }
PulseModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: PulseModule });
PulseModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function PulseModule_Factory(t) { return new (t || PulseModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(PulseModule, { declarations: [PulseComponent], imports: [ɵngcc3.CommonModule], exports: [PulseComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PulseModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    PulseComponent,
                ],
                exports: [
                    PulseComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return PulseModule;
    }());

    var SlidingBarComponent = /** @class */ (function () {
        function SlidingBarComponent() {
            this.colorOne = '#ddd';
            this.colorTwo = '#3c8dbc';
        }
SlidingBarComponent.ɵfac = function SlidingBarComponent_Factory(t) { return new (t || SlidingBarComponent)(); };
SlidingBarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SlidingBarComponent, selectors: [["mk-sliding-bar"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 2, vars: 4, consts: [[1, "sliding-bar"]], template: function SlidingBarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sliding-bar[_ngcontent-%COMP%]{height:4px;overflow:hidden;position:relative;width:100%}.sliding-bar[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:loading 2s linear infinite;background-color:#2980b9;content:\"\";display:block;height:4px;left:-200px;position:absolute;width:200px}@keyframes loading{0%{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlidingBarComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-sliding-bar',
                template: "<div class=\"sliding-bar\" [style.background-color]=\"colorOne\"><div [style.background-color]=\"colorTwo\"></div></div>",
                styles: [".sliding-bar{height:4px;overflow:hidden;position:relative;width:100%}.sliding-bar>div{animation:loading 2s linear infinite;background-color:#2980b9;content:\"\";display:block;height:4px;left:-200px;position:absolute;width:200px}@keyframes loading{0%{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: core.Input
        }], colorTwo: [{
            type: core.Input
        }] }); })();
        return SlidingBarComponent;
    }());
    SlidingBarComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var SlidingBarModule = /** @class */ (function () {
        function SlidingBarModule() {
        }
SlidingBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SlidingBarModule });
SlidingBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SlidingBarModule_Factory(t) { return new (t || SlidingBarModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SlidingBarModule, { declarations: [SlidingBarComponent], imports: [ɵngcc3.CommonModule], exports: [SlidingBarComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlidingBarModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    SlidingBarComponent,
                ],
                exports: [
                    SlidingBarComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return SlidingBarModule;
    }());

    var ThreeBounceComponent = /** @class */ (function () {
        function ThreeBounceComponent() {
            this.colorOne = '#00a65a';
            this.colorTwo = '#fdba2c';
            this.colorThree = '#da4733';
        }
ThreeBounceComponent.ɵfac = function ThreeBounceComponent_Factory(t) { return new (t || ThreeBounceComponent)(); };
ThreeBounceComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ThreeBounceComponent, selectors: [["mk-three-bounce"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo", colorThree: "colorThree" }, decls: 4, vars: 6, consts: [[1, "sk-three-bounce"], [1, "sk-child", "sk-bounce1"], [1, "sk-child", "sk-bounce2"], [1, "sk-child", "sk-bounce3"]], template: function ThreeBounceComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorThree);
    } }, styles: [".sk-three-bounce[_ngcontent-%COMP%]{margin:40px auto;text-align:center;width:80px}.sk-three-bounce[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{animation:sk-three-bounce 1.4s ease-in-out 0s infinite both;background-color:#333;border-radius:100%;display:inline-block;height:20px;width:20px}.sk-three-bounce[_ngcontent-%COMP%]   .sk-bounce1[_ngcontent-%COMP%]{animation-delay:-.32s}.sk-three-bounce[_ngcontent-%COMP%]   .sk-bounce2[_ngcontent-%COMP%]{animation-delay:-.16s}@keyframes sk-three-bounce{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ThreeBounceComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-three-bounce',
                template: "\n    <div class=\"sk-three-bounce\">\n       <div class=\"sk-child sk-bounce1\" [style.background-color]=\"colorOne\"></div>\n       <div class=\"sk-child sk-bounce2\" [style.background-color]=\"colorTwo\"></div>\n       <div class=\"sk-child sk-bounce3\" [style.background-color]=\"colorThree\"></div>\n     </div>",
                styles: [".sk-three-bounce{margin:40px auto;text-align:center;width:80px}.sk-three-bounce .sk-child{animation:sk-three-bounce 1.4s ease-in-out 0s infinite both;background-color:#333;border-radius:100%;display:inline-block;height:20px;width:20px}.sk-three-bounce .sk-bounce1{animation-delay:-.32s}.sk-three-bounce .sk-bounce2{animation-delay:-.16s}@keyframes sk-three-bounce{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: core.Input
        }], colorTwo: [{
            type: core.Input
        }], colorThree: [{
            type: core.Input
        }] }); })();
        return ThreeBounceComponent;
    }());
    ThreeBounceComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }],
        colorThree: [{ type: core.Input }]
    };

    var ThreeBounceModule = /** @class */ (function () {
        function ThreeBounceModule() {
        }
ThreeBounceModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ThreeBounceModule });
ThreeBounceModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ThreeBounceModule_Factory(t) { return new (t || ThreeBounceModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ThreeBounceModule, { declarations: [ThreeBounceComponent], imports: [ɵngcc3.CommonModule], exports: [ThreeBounceComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ThreeBounceModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    ThreeBounceComponent,
                ],
                exports: [
                    ThreeBounceComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return ThreeBounceModule;
    }());

    var WanderingCubesComponent = /** @class */ (function () {
        function WanderingCubesComponent() {
            this.colorOne = '#da4733';
            this.colorTwo = '#3b78e7';
        }
WanderingCubesComponent.ɵfac = function WanderingCubesComponent_Factory(t) { return new (t || WanderingCubesComponent)(); };
WanderingCubesComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WanderingCubesComponent, selectors: [["mk-wandering-cubes"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-wandering-cubes"], [1, "sk-cube", "sk-cube1"], [1, "sk-cube", "sk-cube2"]], template: function WanderingCubesComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-wandering-cubes[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-wandering-cubes[_ngcontent-%COMP%]   .sk-cube[_ngcontent-%COMP%]{animation:sk-wanderingCube 1.8s ease-in-out -1.8s infinite both;height:10px;left:0;position:absolute;top:0;width:10px}.sk-wandering-cubes[_ngcontent-%COMP%]   .sk-cube2[_ngcontent-%COMP%]{animation-delay:-.9s}@keyframes sk-wanderingCube{0%{transform:rotate(0deg)}25%{transform:translateX(30px) rotate(-90deg) scale(.5)}50%{transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}to{transform:rotate(-1turn)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WanderingCubesComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-wandering-cubes',
                template: "\n    <div class=\"sk-wandering-cubes\">\n      <div class=\"sk-cube sk-cube1\" [style.background-color]=\"colorOne\"></div>\n      <div class=\"sk-cube sk-cube2\" [style.background-color]=\"colorTwo\"></div>\n    </div>",
                styles: [".sk-wandering-cubes{height:40px;margin:40px auto;position:relative;width:40px}.sk-wandering-cubes .sk-cube{animation:sk-wanderingCube 1.8s ease-in-out -1.8s infinite both;height:10px;left:0;position:absolute;top:0;width:10px}.sk-wandering-cubes .sk-cube2{animation-delay:-.9s}@keyframes sk-wanderingCube{0%{transform:rotate(0deg)}25%{transform:translateX(30px) rotate(-90deg) scale(.5)}50%{transform:translateX(30px) translateY(30px) rotate(-179deg)}50.1%{transform:translateX(30px) translateY(30px) rotate(-180deg)}75%{transform:translateX(0) translateY(30px) rotate(-270deg) scale(.5)}to{transform:rotate(-1turn)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: core.Input
        }], colorTwo: [{
            type: core.Input
        }] }); })();
        return WanderingCubesComponent;
    }());
    WanderingCubesComponent.propDecorators = {
        colorOne: [{ type: core.Input }],
        colorTwo: [{ type: core.Input }]
    };

    var WanderingCubesModule = /** @class */ (function () {
        function WanderingCubesModule() {
        }
WanderingCubesModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WanderingCubesModule });
WanderingCubesModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WanderingCubesModule_Factory(t) { return new (t || WanderingCubesModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WanderingCubesModule, { declarations: [WanderingCubesComponent], imports: [ɵngcc3.CommonModule], exports: [WanderingCubesComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WanderingCubesModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    WanderingCubesComponent,
                ],
                exports: [
                    WanderingCubesComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return WanderingCubesModule;
    }());

    var WaveComponent = /** @class */ (function () {
        function WaveComponent() {
            this.color = '#da4733';
        }
WaveComponent.ɵfac = function WaveComponent_Factory(t) { return new (t || WaveComponent)(); };
WaveComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WaveComponent, selectors: [["mk-wave"]], inputs: { color: "color" }, decls: 6, vars: 10, consts: [[1, "sk-wave"], [1, "sk-rect", "sk-rect1"], [1, "sk-rect", "sk-rect2"], [1, "sk-rect", "sk-rect3"], [1, "sk-rect", "sk-rect4"], [1, "sk-rect", "sk-rect5"]], template: function WaveComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelement(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-wave[_ngcontent-%COMP%]{font-size:10px;height:40px;margin:40px auto;text-align:center;width:50px}.sk-wave[_ngcontent-%COMP%]   .sk-rect[_ngcontent-%COMP%]{animation:sk-waveStretchDelay 1.2s ease-in-out infinite;display:inline-block;height:100%;width:6px}.sk-wave[_ngcontent-%COMP%]   .sk-rect1[_ngcontent-%COMP%]{animation-delay:-1.2s}.sk-wave[_ngcontent-%COMP%]   .sk-rect2[_ngcontent-%COMP%]{animation-delay:-1.1s}.sk-wave[_ngcontent-%COMP%]   .sk-rect3[_ngcontent-%COMP%]{animation-delay:-1s}.sk-wave[_ngcontent-%COMP%]   .sk-rect4[_ngcontent-%COMP%]{animation-delay:-.9s}.sk-wave[_ngcontent-%COMP%]   .sk-rect5[_ngcontent-%COMP%]{animation-delay:-.8s}@keyframes sk-waveStretchDelay{0%,40%,to{transform:scaleY(.4)}20%{transform:scaleY(1)}}"] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WaveComponent, [{
        type: core.Component,
        args: [{
                selector: 'mk-wave',
                template: "\n    <div class=\"sk-wave\">\n      <div class=\"sk-rect sk-rect1\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect2\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect3\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect4\" [style.background-color]=\"color\"></div>\n      <div class=\"sk-rect sk-rect5\" [style.background-color]=\"color\"></div>\n    </div>",
                styles: [".sk-wave{font-size:10px;height:40px;margin:40px auto;text-align:center;width:50px}.sk-wave .sk-rect{animation:sk-waveStretchDelay 1.2s ease-in-out infinite;display:inline-block;height:100%;width:6px}.sk-wave .sk-rect1{animation-delay:-1.2s}.sk-wave .sk-rect2{animation-delay:-1.1s}.sk-wave .sk-rect3{animation-delay:-1s}.sk-wave .sk-rect4{animation-delay:-.9s}.sk-wave .sk-rect5{animation-delay:-.8s}@keyframes sk-waveStretchDelay{0%,40%,to{transform:scaleY(.4)}20%{transform:scaleY(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: core.Input
        }] }); })();
        return WaveComponent;
    }());
    WaveComponent.propDecorators = {
        color: [{ type: core.Input }]
    };

    var WaveModule = /** @class */ (function () {
        function WaveModule() {
        }
WaveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WaveModule });
WaveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WaveModule_Factory(t) { return new (t || WaveModule)(); }, imports: [[
            common.CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WaveModule, { declarations: [WaveComponent], imports: [ɵngcc3.CommonModule], exports: [WaveComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WaveModule, [{
        type: core.NgModule,
        args: [{
                imports: [
                    common.CommonModule
                ],
                declarations: [
                    WaveComponent,
                ],
                exports: [
                    WaveComponent
                ]
            }]
    }], function () { return []; }, null); })();
        return WaveModule;
    }());

    /*
     * Public API Surface of angular-loading-page
     */

    /**
     * Generated bundle index. Do not edit.
     */

    exports.ChasingDotsComponent = ChasingDotsComponent;
    exports.ChasingDotsModule = ChasingDotsModule;
    exports.CircleComponent = CircleComponent;
    exports.CircleModule = CircleModule;
    exports.CubeGridComponent = CubeGridComponent;
    exports.CubeGridModule = CubeGridModule;
    exports.DoubleBounceComponent = DoubleBounceComponent;
    exports.DoubleBounceModule = DoubleBounceModule;
    exports.FadingCircleComponent = FadingCircleComponent;
    exports.FadingCircleModule = FadingCircleModule;
    exports.FoldingCubeComponent = FoldingCubeComponent;
    exports.FoldingCubeModule = FoldingCubeModule;
    exports.HttpInterceptorService = HttpInterceptorService;
    exports.HttpInterceptorServiceFactory = HttpInterceptorServiceFactory;
    exports.HttpObservableService = HttpObservableService;
    exports.LoadingPageDirective = LoadingPageDirective;
    exports.LoadingPageModule = LoadingPageModule;
    exports.MaterialBarModule = MaterialBarModule;
    exports.MaterialbarComponent = MaterialbarComponent;
    exports.PulseComponent = PulseComponent;
    exports.PulseModule = PulseModule;
    exports.RouteInterceptorService = RouteInterceptorService;
    exports.SlidingBarComponent = SlidingBarComponent;
    exports.SlidingBarModule = SlidingBarModule;
    exports.ThreeBounceComponent = ThreeBounceComponent;
    exports.ThreeBounceModule = ThreeBounceModule;
    exports.WanderingCubesComponent = WanderingCubesComponent;
    exports.WanderingCubesModule = WanderingCubesModule;
    exports.WaveComponent = WaveComponent;
    exports.WaveModule = WaveModule;
    exports.ɵ0 = ɵ0;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
