// angular-loading-page/esm2015/lib/loaders/fading-circle/fading-circle.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class FadingCircleComponent {
    constructor() {
        this.color = '#da4733';
    }
}
FadingCircleComponent.ɵfac = function FadingCircleComponent_Factory(t) { return new (t || FadingCircleComponent)(); };
FadingCircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: FadingCircleComponent, selectors: [["mk-fading-circle"]], inputs: { color: "color" }, decls: 25, vars: 24, consts: [[1, "sk-fading-circle"], [1, "sk-circle1", "sk-circle"], [1, "sk-circle2", "sk-circle"], [1, "sk-circle3", "sk-circle"], [1, "sk-circle4", "sk-circle"], [1, "sk-circle5", "sk-circle"], [1, "sk-circle6", "sk-circle"], [1, "sk-circle7", "sk-circle"], [1, "sk-circle8", "sk-circle"], [1, "sk-circle9", "sk-circle"], [1, "sk-circle10", "sk-circle"], [1, "sk-circle11", "sk-circle"], [1, "sk-circle12", "sk-circle"]], template: function FadingCircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 5);
        ɵngcc0.ɵɵelement(10, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 6);
        ɵngcc0.ɵɵelement(12, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 7);
        ɵngcc0.ɵɵelement(14, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "div", 8);
        ɵngcc0.ɵɵelement(16, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div", 9);
        ɵngcc0.ɵɵelement(18, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 10);
        ɵngcc0.ɵɵelement(20, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 11);
        ɵngcc0.ɵɵelement(22, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(23, "div", 12);
        ɵngcc0.ɵɵelement(24, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-fading-circle[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle[_ngcontent-%COMP%]{height:100%;left:0;position:absolute;top:0;width:100%}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:sk-circleFadeDelay 1.2s ease-in-out infinite both;border-radius:100%;content:\"\";display:block;height:15%;margin:0 auto;width:15%}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%]{transform:rotate(30deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%]{transform:rotate(60deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%]{transform:rotate(90deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%]{transform:rotate(120deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%]{transform:rotate(150deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%]{transform:rotate(180deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%]{transform:rotate(210deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%]{transform:rotate(240deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%]{transform:rotate(270deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%]{transform:rotate(300deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%]{transform:rotate(330deg)}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1.1s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.9s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.8s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.7s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.6s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.5s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.4s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.3s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.2s}.sk-fading-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.1s}@keyframes sk-circleFadeDelay{0%,39%,to{opacity:0}40%{opacity:1}}"] });
FadingCircleComponent.propDecorators = {
    color: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FadingCircleComponent, [{
        type: Component,
        args: [{
                selector: 'mk-fading-circle',
                template: `
    <div class="sk-fading-circle">
      <div class="sk-circle1 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle2 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle3 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle4 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle5 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle6 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle7 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle8 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle9 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle10 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle11 sk-circle"><div [style.background-color]="color"></div></div>
      <div class="sk-circle12 sk-circle"><div [style.background-color]="color"></div></div>
    </div>`,
                styles: [".sk-fading-circle{height:40px;margin:40px auto;position:relative;width:40px}.sk-fading-circle .sk-circle{height:100%;left:0;position:absolute;top:0;width:100%}.sk-fading-circle .sk-circle>div{animation:sk-circleFadeDelay 1.2s ease-in-out infinite both;border-radius:100%;content:\"\";display:block;height:15%;margin:0 auto;width:15%}.sk-fading-circle .sk-circle2{transform:rotate(30deg)}.sk-fading-circle .sk-circle3{transform:rotate(60deg)}.sk-fading-circle .sk-circle4{transform:rotate(90deg)}.sk-fading-circle .sk-circle5{transform:rotate(120deg)}.sk-fading-circle .sk-circle6{transform:rotate(150deg)}.sk-fading-circle .sk-circle7{transform:rotate(180deg)}.sk-fading-circle .sk-circle8{transform:rotate(210deg)}.sk-fading-circle .sk-circle9{transform:rotate(240deg)}.sk-fading-circle .sk-circle10{transform:rotate(270deg)}.sk-fading-circle .sk-circle11{transform:rotate(300deg)}.sk-fading-circle .sk-circle12{transform:rotate(330deg)}.sk-fading-circle .sk-circle2>div{animation-delay:-1.1s}.sk-fading-circle .sk-circle3>div{animation-delay:-1s}.sk-fading-circle .sk-circle4>div{animation-delay:-.9s}.sk-fading-circle .sk-circle5>div{animation-delay:-.8s}.sk-fading-circle .sk-circle6>div{animation-delay:-.7s}.sk-fading-circle .sk-circle7>div{animation-delay:-.6s}.sk-fading-circle .sk-circle8>div{animation-delay:-.5s}.sk-fading-circle .sk-circle9>div{animation-delay:-.4s}.sk-fading-circle .sk-circle10>div{animation-delay:-.3s}.sk-fading-circle .sk-circle11>div{animation-delay:-.2s}.sk-fading-circle .sk-circle12>div{animation-delay:-.1s}@keyframes sk-circleFadeDelay{0%,39%,to{opacity:0}40%{opacity:1}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
