// angular-loading-page/esm2015/lib/loaders/fading-circle/fading-circle.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FadingCircleComponent } from './fading-circle.component';
import * as ɵngcc0 from '@angular/core';
export class FadingCircleModule {
}
FadingCircleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FadingCircleModule });
FadingCircleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FadingCircleModule_Factory(t) { return new (t || FadingCircleModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FadingCircleModule, { declarations: function () { return [FadingCircleComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [FadingCircleComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FadingCircleModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    FadingCircleComponent,
                ],
                exports: [
                    FadingCircleComponent
                ]
            }]
    }], null, null); })();
