// angular-loading-page/esm2015/lib/loaders/material-bar/material-bar.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class MaterialbarComponent {
    constructor() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
        this.colorThree = '#fdba2c';
    }
}
MaterialbarComponent.ɵfac = function MaterialbarComponent_Factory(t) { return new (t || MaterialbarComponent)(); };
MaterialbarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: MaterialbarComponent, selectors: [["mk-material-bar"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo", colorThree: "colorThree" }, decls: 4, vars: 6, consts: [[1, "material-bar"], [1, "bar"]], template: function MaterialbarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 1);
        ɵngcc0.ɵɵelement(3, "div", 1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorThree);
    } }, styles: [".material-bar[_ngcontent-%COMP%]{background-color:#fdba2c;height:4px;position:relative;width:100%}.bar[_ngcontent-%COMP%]{content:\"\";display:inline;height:100%;left:50%;position:absolute;text-align:center;width:0}.bar[_ngcontent-%COMP%]:first-child{animation:loading 3s linear infinite}.bar[_ngcontent-%COMP%]:nth-child(2){animation:loading 3s linear 1s infinite}.bar[_ngcontent-%COMP%]:nth-child(3){animation:loading 3s linear 2s infinite}@keyframes loading{0%{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"] });
MaterialbarComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }],
    colorThree: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialbarComponent, [{
        type: Component,
        args: [{
                selector: 'mk-material-bar',
                template: `
  <div class="material-bar">
    <div class="bar" [style.background-color]="colorOne"></div>
    <div class="bar" [style.background-color]="colorTwo"></div>
    <div class="bar" [style.background-color]="colorThree"></div>
  </div>`,
                styles: [".material-bar{background-color:#fdba2c;height:4px;position:relative;width:100%}.bar{content:\"\";display:inline;height:100%;left:50%;position:absolute;text-align:center;width:0}.bar:first-child{animation:loading 3s linear infinite}.bar:nth-child(2){animation:loading 3s linear 1s infinite}.bar:nth-child(3){animation:loading 3s linear 2s infinite}@keyframes loading{0%{left:50%;width:0;z-index:100}33.3333%{left:0;width:100%;z-index:10}to{left:0;width:100%}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }], colorThree: [{
            type: Input
        }] }); })();
