// angular-loading-page/esm2015/lib/loaders/material-bar/material-bar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { MaterialbarComponent } from './material-bar.component';
import * as ɵngcc0 from '@angular/core';
export class MaterialBarModule {
}
MaterialBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MaterialBarModule });
MaterialBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MaterialBarModule_Factory(t) { return new (t || MaterialBarModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MaterialBarModule, { declarations: function () { return [MaterialbarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [MaterialbarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MaterialBarModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    MaterialbarComponent,
                ],
                exports: [
                    MaterialbarComponent
                ]
            }]
    }], null, null); })();
