// angular-admin-lte/esm2015/lib/tabs/tabs.directive.js
import { Directive, Input, ElementRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class TabToggleDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
    }
}
TabToggleDirective.ɵfac = function TabToggleDirective_Factory(t) { return new (t || TabToggleDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
TabToggleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: TabToggleDirective, selectors: [["", "mkTabToggle", ""]], inputs: { tabComponent: ["mkTabToggle", "tabComponent"] } });
TabToggleDirective.ctorParameters = () => [
    { type: ElementRef }
];
TabToggleDirective.propDecorators = {
    tabComponent: [{ type: Input, args: ['mkTabToggle',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TabToggleDirective, [{
        type: Directive,
        args: [{
                selector: '[mkTabToggle]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { tabComponent: [{
            type: Input,
            args: ['mkTabToggle']
        }] }); })();
