// angular-admin-lte/esm2015/lib/box-small/box-small.directive.js
import { Directive } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class BoxSmallFooterDirective {
}
BoxSmallFooterDirective.ɵfac = function BoxSmallFooterDirective_Factory(t) { return new (t || BoxSmallFooterDirective)(); };
BoxSmallFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxSmallFooterDirective, selectors: [["mk-box-small-footer"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxSmallFooterDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-small-footer'
            }]
    }], null, null); })();
export class BoxSmallHeaderDirective {
}
BoxSmallHeaderDirective.ɵfac = function BoxSmallHeaderDirective_Factory(t) { return new (t || BoxSmallHeaderDirective)(); };
BoxSmallHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxSmallHeaderDirective, selectors: [["mk-box-small-header"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxSmallHeaderDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-small-header'
            }]
    }], null, null); })();
export class BoxSmallContentDirective {
}
BoxSmallContentDirective.ɵfac = function BoxSmallContentDirective_Factory(t) { return new (t || BoxSmallContentDirective)(); };
BoxSmallContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxSmallContentDirective, selectors: [["mk-box-small-content"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxSmallContentDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-small-content'
            }]
    }], null, null); })();
