// angular-admin-lte/esm2015/lib/box-small/box-small.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ColorModule } from '../color/color.module';
import { BoxSmallHeaderDirective, BoxSmallContentDirective, BoxSmallFooterDirective } from './box-small.directive';
import { BoxSmallComponent } from './box-small.component';
import * as ɵngcc0 from '@angular/core';
export { BoxSmallHeaderDirective, BoxSmallContentDirective, BoxSmallFooterDirective } from './box-small.directive';
export { BoxSmallComponent } from './box-small.component';
export class BoxSmallModule {
}
BoxSmallModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BoxSmallModule });
BoxSmallModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BoxSmallModule_Factory(t) { return new (t || BoxSmallModule)(); }, imports: [[CommonModule, ColorModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BoxSmallModule, { declarations: function () { return [BoxSmallComponent, BoxSmallHeaderDirective, BoxSmallContentDirective, BoxSmallFooterDirective]; }, imports: function () { return [CommonModule, ColorModule]; }, exports: function () { return [BoxSmallComponent, BoxSmallHeaderDirective, BoxSmallContentDirective, BoxSmallFooterDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxSmallModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, ColorModule],
                exports: [BoxSmallComponent, BoxSmallHeaderDirective, BoxSmallContentDirective, BoxSmallFooterDirective],
                declarations: [BoxSmallComponent, BoxSmallHeaderDirective, BoxSmallContentDirective, BoxSmallFooterDirective]
            }]
    }], null, null); })();
