// angular-admin-lte/esm2015/lib/layout/layout.service.js
import { Injectable } from '@angular/core';
import { ActivationStart, Router } from '@angular/router';
import { BehaviorSubject } from 'rxjs';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/router';
export class LayoutService {
    constructor(router) {
        this.router = router;
        this.isCustomLayout = new BehaviorSubject(false);
        this.customLayout = false;
        this.init();
    }
    init() {
        this.router.events.subscribe(event => {
            if (event instanceof ActivationStart) {
                this.customLayout = event.snapshot.data.customLayout;
                this.isCustomLayout.next(this.customLayout);
            }
        });
    }
}
LayoutService.ɵfac = function LayoutService_Factory(t) { return new (t || LayoutService)(ɵngcc0.ɵɵinject(ɵngcc1.Router)); };
LayoutService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: LayoutService, factory: LayoutService.ɵfac });
LayoutService.ctorParameters = () => [
    { type: Router }
];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LayoutService, [{
        type: Injectable
    }], function () { return [{ type: ɵngcc1.Router }]; }, null); })();
