// angular-loading-page/esm2015/lib/loaders/double-bounce/double-bounce.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class DoubleBounceComponent {
    constructor() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
}
DoubleBounceComponent.ɵfac = function DoubleBounceComponent_Factory(t) { return new (t || DoubleBounceComponent)(); };
DoubleBounceComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DoubleBounceComponent, selectors: [["mk-double-bounce"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 3, vars: 4, consts: [[1, "sk-double-bounce"], [1, "sk-child", "sk-double-bounce1"], [1, "sk-child", "sk-double-bounce2"]], template: function DoubleBounceComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sk-double-bounce[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-double-bounce[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{animation:sk-doubleBounce 2s ease-in-out infinite;border-radius:50%;height:100%;left:0;opacity:.6;position:absolute;top:0;width:100%}.sk-double-bounce[_ngcontent-%COMP%]   .sk-double-bounce2[_ngcontent-%COMP%]{animation-delay:-1s}@keyframes sk-doubleBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"] });
DoubleBounceComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DoubleBounceComponent, [{
        type: Component,
        args: [{
                selector: 'mk-double-bounce',
                template: `
    <div class="sk-double-bounce">
      <div class="sk-child sk-double-bounce1" [style.background-color]="colorOne"></div>
      <div class="sk-child sk-double-bounce2" [style.background-color]="colorTwo"></div>
    </div>`,
                styles: [".sk-double-bounce{height:40px;margin:40px auto;position:relative;width:40px}.sk-double-bounce .sk-child{animation:sk-doubleBounce 2s ease-in-out infinite;border-radius:50%;height:100%;left:0;opacity:.6;position:absolute;top:0;width:100%}.sk-double-bounce .sk-double-bounce2{animation-delay:-1s}@keyframes sk-doubleBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
