// angular-loading-page/esm2015/lib/loaders/double-bounce/double-bounce.component.js
import { Component, Input } from '@angular/core';
export class DoubleBounceComponent {
    constructor() {
        this.colorOne = '#da4733';
        this.colorTwo = '#3b78e7';
    }
}
DoubleBounceComponent.decorators = [
    { type: Component, args: [{
                selector: 'mk-double-bounce',
                template: `
    <div class="sk-double-bounce">
      <div class="sk-child sk-double-bounce1" [style.background-color]="colorOne"></div>
      <div class="sk-child sk-double-bounce2" [style.background-color]="colorTwo"></div>
    </div>`,
                styles: [".sk-double-bounce{height:40px;margin:40px auto;position:relative;width:40px}.sk-double-bounce .sk-child{animation:sk-doubleBounce 2s ease-in-out infinite;border-radius:50%;height:100%;left:0;opacity:.6;position:absolute;top:0;width:100%}.sk-double-bounce .sk-double-bounce2{animation-delay:-1s}@keyframes sk-doubleBounce{0%,to{transform:scale(0)}50%{transform:scale(1)}}"]
            },] }
];
DoubleBounceComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }]
};
