// angular-loading-page/esm2015/lib/loaders/double-bounce/double-bounce.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DoubleBounceComponent } from './double-bounce.component';
import * as ɵngcc0 from '@angular/core';
export class DoubleBounceModule {
}
DoubleBounceModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DoubleBounceModule });
DoubleBounceModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DoubleBounceModule_Factory(t) { return new (t || DoubleBounceModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DoubleBounceModule, { declarations: function () { return [DoubleBounceComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [DoubleBounceComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DoubleBounceModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    DoubleBounceComponent,
                ],
                exports: [
                    DoubleBounceComponent
                ]
            }]
    }], null, null); })();
