// angular-admin-lte/esm2015/lib/box-info/box-info.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ColorModule } from '../color/color.module';
import { BoxInfoHeaderDirective, BoxInfoContentDirective, BoxInfoFooterDirective } from './box-info.directive';
import { BoxInfoComponent } from './box-info.component';
import * as ɵngcc0 from '@angular/core';
export { BoxInfoHeaderDirective, BoxInfoContentDirective, BoxInfoFooterDirective } from './box-info.directive';
export { BoxInfoComponent } from './box-info.component';
export class BoxInfoModule {
}
BoxInfoModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BoxInfoModule });
BoxInfoModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BoxInfoModule_Factory(t) { return new (t || BoxInfoModule)(); }, imports: [[CommonModule, ColorModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BoxInfoModule, { declarations: function () { return [BoxInfoComponent, BoxInfoHeaderDirective, BoxInfoContentDirective, BoxInfoFooterDirective]; }, imports: function () { return [CommonModule, ColorModule]; }, exports: function () { return [BoxInfoComponent, BoxInfoHeaderDirective, BoxInfoContentDirective, BoxInfoFooterDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxInfoModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, ColorModule],
                exports: [BoxInfoComponent, BoxInfoHeaderDirective, BoxInfoContentDirective, BoxInfoFooterDirective],
                declarations: [BoxInfoComponent, BoxInfoHeaderDirective, BoxInfoContentDirective, BoxInfoFooterDirective]
            }]
    }], null, null); })();
