// angular-admin-lte/esm2015/lib/box-info/box-info.directive.js
import { Directive } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class BoxInfoContentDirective {
}
BoxInfoContentDirective.ɵfac = function BoxInfoContentDirective_Factory(t) { return new (t || BoxInfoContentDirective)(); };
BoxInfoContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxInfoContentDirective, selectors: [["mk-box-info-content"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxInfoContentDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-info-content'
            }]
    }], null, null); })();
export class BoxInfoFooterDirective {
}
BoxInfoFooterDirective.ɵfac = function BoxInfoFooterDirective_Factory(t) { return new (t || BoxInfoFooterDirective)(); };
BoxInfoFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxInfoFooterDirective, selectors: [["mk-box-info-footer"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxInfoFooterDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-info-footer'
            }]
    }], null, null); })();
export class BoxInfoHeaderDirective {
}
BoxInfoHeaderDirective.ɵfac = function BoxInfoHeaderDirective_Factory(t) { return new (t || BoxInfoHeaderDirective)(); };
BoxInfoHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxInfoHeaderDirective, selectors: [["mk-box-info-header"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxInfoHeaderDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-info-header'
            }]
    }], null, null); })();
