// angular-admin-lte/esm2015/lib/box/box.directive.js
import { Directive } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class BoxContentDirective {
}
BoxContentDirective.ɵfac = function BoxContentDirective_Factory(t) { return new (t || BoxContentDirective)(); };
BoxContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxContentDirective, selectors: [["mk-box-content"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxContentDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-content'
            }]
    }], null, null); })();
export class BoxFooterDirective {
}
BoxFooterDirective.ɵfac = function BoxFooterDirective_Factory(t) { return new (t || BoxFooterDirective)(); };
BoxFooterDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxFooterDirective, selectors: [["mk-box-footer"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxFooterDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-footer'
            }]
    }], null, null); })();
export class BoxToolsDirective {
}
BoxToolsDirective.ɵfac = function BoxToolsDirective_Factory(t) { return new (t || BoxToolsDirective)(); };
BoxToolsDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxToolsDirective, selectors: [["mk-box-tools"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxToolsDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-tools'
            }]
    }], null, null); })();
export class BoxHeaderDirective {
}
BoxHeaderDirective.ɵfac = function BoxHeaderDirective_Factory(t) { return new (t || BoxHeaderDirective)(); };
BoxHeaderDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: BoxHeaderDirective, selectors: [["mk-box-header"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxHeaderDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-box-header'
            }]
    }], null, null); })();
