// angular-admin-lte/esm2015/lib/box/box.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AnimationsModule } from '../animations/animations.module';
import { ColorModule } from '../color/color.module';
import { BoxHeaderDirective, BoxContentDirective, BoxFooterDirective, BoxToolsDirective } from './box.directive';
import { BoxComponent } from './box.component';
import * as ɵngcc0 from '@angular/core';
export { BoxHeaderDirective, BoxContentDirective, BoxFooterDirective, BoxToolsDirective } from './box.directive';
export { BoxComponent } from './box.component';
export class BoxModule {
}
BoxModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: BoxModule });
BoxModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function BoxModule_Factory(t) { return new (t || BoxModule)(); }, imports: [[CommonModule, AnimationsModule, ColorModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(BoxModule, { declarations: function () { return [BoxComponent, BoxHeaderDirective, BoxContentDirective, BoxFooterDirective, BoxToolsDirective]; }, imports: function () { return [CommonModule, AnimationsModule, ColorModule]; }, exports: function () { return [BoxComponent, BoxHeaderDirective, BoxContentDirective, BoxFooterDirective, BoxToolsDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(BoxModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, AnimationsModule, ColorModule],
                exports: [BoxComponent, BoxHeaderDirective, BoxContentDirective, BoxFooterDirective, BoxToolsDirective],
                declarations: [BoxComponent, BoxHeaderDirective, BoxContentDirective, BoxFooterDirective, BoxToolsDirective]
            }]
    }], null, null); })();
