// angular-admin-lte/esm2015/lib/form/input-text/input-text.module.js
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { ColorModule } from '../../color/color.module';
import { InputTextDirective } from './input-text.directive';
import * as ɵngcc0 from '@angular/core';
export { InputTextDirective } from './input-text.directive';
export class InputTextModule {
}
InputTextModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: InputTextModule });
InputTextModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function InputTextModule_Factory(t) { return new (t || InputTextModule)(); }, imports: [[
            CommonModule,
            ColorModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(InputTextModule, { declarations: function () { return [InputTextDirective]; }, imports: function () { return [CommonModule,
        ColorModule,
        FormsModule]; }, exports: function () { return [InputTextDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputTextModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    ColorModule,
                    FormsModule
                ],
                exports: [InputTextDirective],
                declarations: [InputTextDirective]
            }]
    }], null, null); })();
