// angular-admin-lte/esm2015/lib/dropdown/dropdown.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AnimationsModule } from '../animations/animations.module';
import { ColorModule } from '../color/color.module';
import { DropdownComponent, DropdownToggleComponent, DropdownMenuComponent } from './dropdown.component';
import * as ɵngcc0 from '@angular/core';
export { DropdownComponent, DropdownToggleComponent, DropdownMenuComponent } from './dropdown.component';
export class DropdownModule {
}
DropdownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DropdownModule });
DropdownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DropdownModule_Factory(t) { return new (t || DropdownModule)(); }, imports: [[CommonModule, AnimationsModule, ColorModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DropdownModule, { declarations: function () { return [DropdownComponent, DropdownToggleComponent, DropdownMenuComponent]; }, imports: function () { return [CommonModule, AnimationsModule, ColorModule]; }, exports: function () { return [DropdownComponent, DropdownToggleComponent, DropdownMenuComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DropdownModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, AnimationsModule, ColorModule],
                exports: [DropdownComponent, DropdownToggleComponent, DropdownMenuComponent],
                declarations: [DropdownComponent, DropdownToggleComponent, DropdownMenuComponent]
            }]
    }], null, null); })();
