// angular-admin-lte/esm2015/lib/layout/header/header.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { HeaderComponent, HeaderLogoComponent, HeaderLogoMiniComponent } from './header.component';
import * as ɵngcc0 from '@angular/core';
export { HeaderComponent, HeaderLogoComponent, HeaderLogoMiniComponent } from './header.component';
export class HeaderModule {
}
HeaderModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: HeaderModule });
HeaderModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function HeaderModule_Factory(t) { return new (t || HeaderModule)(); }, imports: [[CommonModule, RouterModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(HeaderModule, { declarations: function () { return [HeaderComponent, HeaderLogoComponent, HeaderLogoMiniComponent]; }, imports: function () { return [CommonModule, RouterModule]; }, exports: function () { return [HeaderComponent, HeaderLogoComponent, HeaderLogoMiniComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(HeaderModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule],
                exports: [HeaderComponent, HeaderLogoComponent, HeaderLogoMiniComponent],
                declarations: [HeaderComponent, HeaderLogoComponent, HeaderLogoMiniComponent]
            }]
    }], null, null); })();
