// angular-admin-lte/esm2015/lib/layout/sidebar-right/sidebar-right.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SidebarRightComponent } from './sidebar-right.component';
import * as ɵngcc0 from '@angular/core';
export { SidebarRightComponent } from './sidebar-right.component';
export class SidebarRightModule {
}
SidebarRightModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SidebarRightModule });
SidebarRightModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SidebarRightModule_Factory(t) { return new (t || SidebarRightModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SidebarRightModule, { declarations: function () { return [SidebarRightComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SidebarRightComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SidebarRightModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [SidebarRightComponent],
                declarations: [SidebarRightComponent]
            }]
    }], null, null); })();
