// angular-loading-page/esm2015/lib/loaders/circle/circle.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class CircleComponent {
    constructor() {
        this.color = '#da4733';
    }
}
CircleComponent.ɵfac = function CircleComponent_Factory(t) { return new (t || CircleComponent)(); };
CircleComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CircleComponent, selectors: [["mk-circle"]], inputs: { color: "color" }, decls: 25, vars: 24, consts: [[1, "sk-circle"], [1, "sk-circle1", "sk-child"], [1, "sk-circle2", "sk-child"], [1, "sk-circle3", "sk-child"], [1, "sk-circle4", "sk-child"], [1, "sk-circle5", "sk-child"], [1, "sk-circle6", "sk-child"], [1, "sk-circle7", "sk-child"], [1, "sk-circle8", "sk-child"], [1, "sk-circle9", "sk-child"], [1, "sk-circle10", "sk-child"], [1, "sk-circle11", "sk-child"], [1, "sk-circle12", "sk-child"]], template: function CircleComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelement(4, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵelement(6, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "div", 4);
        ɵngcc0.ɵɵelement(8, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "div", 5);
        ɵngcc0.ɵɵelement(10, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 6);
        ɵngcc0.ɵɵelement(12, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 7);
        ɵngcc0.ɵɵelement(14, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(15, "div", 8);
        ɵngcc0.ɵɵelement(16, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div", 9);
        ɵngcc0.ɵɵelement(18, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div", 10);
        ɵngcc0.ɵɵelement(20, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div", 11);
        ɵngcc0.ɵɵelement(22, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(23, "div", 12);
        ɵngcc0.ɵɵelement(24, "div");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-circle[_ngcontent-%COMP%]{height:40px;margin:40px auto;position:relative;width:40px}.sk-circle[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%]{height:100%;left:0;position:absolute;top:0;width:100%}.sk-circle[_ngcontent-%COMP%]   .sk-child[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:sk-circleBounceDelay 1.2s ease-in-out infinite both;border-radius:100%;display:block;height:15%;margin:0 auto;width:15%}.sk-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%]{transform:rotate(30deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%]{transform:rotate(60deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%]{transform:rotate(90deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%]{transform:rotate(120deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%]{transform:rotate(150deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%]{transform:rotate(180deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%]{transform:rotate(210deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%]{transform:rotate(240deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%]{transform:rotate(270deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%]{transform:rotate(300deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%]{transform:rotate(330deg)}.sk-circle[_ngcontent-%COMP%]   .sk-circle2[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1.1s}.sk-circle[_ngcontent-%COMP%]   .sk-circle3[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-1s}.sk-circle[_ngcontent-%COMP%]   .sk-circle4[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.9s}.sk-circle[_ngcontent-%COMP%]   .sk-circle5[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.8s}.sk-circle[_ngcontent-%COMP%]   .sk-circle6[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.7s}.sk-circle[_ngcontent-%COMP%]   .sk-circle7[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.6s}.sk-circle[_ngcontent-%COMP%]   .sk-circle8[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.5s}.sk-circle[_ngcontent-%COMP%]   .sk-circle9[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.4s}.sk-circle[_ngcontent-%COMP%]   .sk-circle10[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.3s}.sk-circle[_ngcontent-%COMP%]   .sk-circle11[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.2s}.sk-circle[_ngcontent-%COMP%]   .sk-circle12[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation-delay:-.1s}@keyframes sk-circleBounceDelay{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"] });
CircleComponent.propDecorators = {
    color: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleComponent, [{
        type: Component,
        args: [{
                selector: 'mk-circle',
                template: `
    <div class="sk-circle">
      <div class="sk-circle1 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle2 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle3 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle4 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle5 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle6 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle7 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle8 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle9 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle10 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle11 sk-child"><div [style.background-color]="color"></div></div>
      <div class="sk-circle12 sk-child"><div [style.background-color]="color"></div></div>
    </div>`,
                styles: [".sk-circle{height:40px;margin:40px auto;position:relative;width:40px}.sk-circle .sk-child{height:100%;left:0;position:absolute;top:0;width:100%}.sk-circle .sk-child>div{animation:sk-circleBounceDelay 1.2s ease-in-out infinite both;border-radius:100%;display:block;height:15%;margin:0 auto;width:15%}.sk-circle .sk-circle2{transform:rotate(30deg)}.sk-circle .sk-circle3{transform:rotate(60deg)}.sk-circle .sk-circle4{transform:rotate(90deg)}.sk-circle .sk-circle5{transform:rotate(120deg)}.sk-circle .sk-circle6{transform:rotate(150deg)}.sk-circle .sk-circle7{transform:rotate(180deg)}.sk-circle .sk-circle8{transform:rotate(210deg)}.sk-circle .sk-circle9{transform:rotate(240deg)}.sk-circle .sk-circle10{transform:rotate(270deg)}.sk-circle .sk-circle11{transform:rotate(300deg)}.sk-circle .sk-circle12{transform:rotate(330deg)}.sk-circle .sk-circle2>div{animation-delay:-1.1s}.sk-circle .sk-circle3>div{animation-delay:-1s}.sk-circle .sk-circle4>div{animation-delay:-.9s}.sk-circle .sk-circle5>div{animation-delay:-.8s}.sk-circle .sk-circle6>div{animation-delay:-.7s}.sk-circle .sk-circle7>div{animation-delay:-.6s}.sk-circle .sk-circle8>div{animation-delay:-.5s}.sk-circle .sk-circle9>div{animation-delay:-.4s}.sk-circle .sk-circle10>div{animation-delay:-.3s}.sk-circle .sk-circle11>div{animation-delay:-.2s}.sk-circle .sk-circle12>div{animation-delay:-.1s}@keyframes sk-circleBounceDelay{0%,80%,to{transform:scale(0)}40%{transform:scale(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
