// angular-loading-page/esm2015/lib/loaders/circle/circle.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CircleComponent } from './circle.component';
import * as ɵngcc0 from '@angular/core';
export class CircleModule {
}
CircleModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CircleModule });
CircleModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CircleModule_Factory(t) { return new (t || CircleModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CircleModule, { declarations: function () { return [CircleComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CircleComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CircleModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    CircleComponent,
                ],
                exports: [
                    CircleComponent
                ]
            }]
    }], null, null); })();
