// angular-loading-page/esm2015/lib/loaders/wave/wave.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { WaveComponent } from './wave.component';
import * as ɵngcc0 from '@angular/core';
export class WaveModule {
}
WaveModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: WaveModule });
WaveModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function WaveModule_Factory(t) { return new (t || WaveModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(WaveModule, { declarations: function () { return [WaveComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [WaveComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WaveModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    WaveComponent,
                ],
                exports: [
                    WaveComponent
                ]
            }]
    }], null, null); })();
