// angular-loading-page/esm2015/lib/loaders/wave/wave.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class WaveComponent {
    constructor() {
        this.color = '#da4733';
    }
}
WaveComponent.ɵfac = function WaveComponent_Factory(t) { return new (t || WaveComponent)(); };
WaveComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: WaveComponent, selectors: [["mk-wave"]], inputs: { color: "color" }, decls: 6, vars: 10, consts: [[1, "sk-wave"], [1, "sk-rect", "sk-rect1"], [1, "sk-rect", "sk-rect2"], [1, "sk-rect", "sk-rect3"], [1, "sk-rect", "sk-rect4"], [1, "sk-rect", "sk-rect5"]], template: function WaveComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div", 1);
        ɵngcc0.ɵɵelement(2, "div", 2);
        ɵngcc0.ɵɵelement(3, "div", 3);
        ɵngcc0.ɵɵelement(4, "div", 4);
        ɵngcc0.ɵɵelement(5, "div", 5);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.color);
    } }, styles: [".sk-wave[_ngcontent-%COMP%]{font-size:10px;height:40px;margin:40px auto;text-align:center;width:50px}.sk-wave[_ngcontent-%COMP%]   .sk-rect[_ngcontent-%COMP%]{animation:sk-waveStretchDelay 1.2s ease-in-out infinite;display:inline-block;height:100%;width:6px}.sk-wave[_ngcontent-%COMP%]   .sk-rect1[_ngcontent-%COMP%]{animation-delay:-1.2s}.sk-wave[_ngcontent-%COMP%]   .sk-rect2[_ngcontent-%COMP%]{animation-delay:-1.1s}.sk-wave[_ngcontent-%COMP%]   .sk-rect3[_ngcontent-%COMP%]{animation-delay:-1s}.sk-wave[_ngcontent-%COMP%]   .sk-rect4[_ngcontent-%COMP%]{animation-delay:-.9s}.sk-wave[_ngcontent-%COMP%]   .sk-rect5[_ngcontent-%COMP%]{animation-delay:-.8s}@keyframes sk-waveStretchDelay{0%,40%,to{transform:scaleY(.4)}20%{transform:scaleY(1)}}"] });
WaveComponent.propDecorators = {
    color: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(WaveComponent, [{
        type: Component,
        args: [{
                selector: 'mk-wave',
                template: `
    <div class="sk-wave">
      <div class="sk-rect sk-rect1" [style.background-color]="color"></div>
      <div class="sk-rect sk-rect2" [style.background-color]="color"></div>
      <div class="sk-rect sk-rect3" [style.background-color]="color"></div>
      <div class="sk-rect sk-rect4" [style.background-color]="color"></div>
      <div class="sk-rect sk-rect5" [style.background-color]="color"></div>
    </div>`,
                styles: [".sk-wave{font-size:10px;height:40px;margin:40px auto;text-align:center;width:50px}.sk-wave .sk-rect{animation:sk-waveStretchDelay 1.2s ease-in-out infinite;display:inline-block;height:100%;width:6px}.sk-wave .sk-rect1{animation-delay:-1.2s}.sk-wave .sk-rect2{animation-delay:-1.1s}.sk-wave .sk-rect3{animation-delay:-1s}.sk-wave .sk-rect4{animation-delay:-.9s}.sk-wave .sk-rect5{animation-delay:-.8s}@keyframes sk-waveStretchDelay{0%,40%,to{transform:scaleY(.4)}20%{transform:scaleY(1)}}"]
            }]
    }], function () { return []; }, { color: [{
            type: Input
        }] }); })();
