// angular-loading-page/esm2015/lib/loaders/sliding-bar/sliding-bar.component.js
import { Component, Input } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class SlidingBarComponent {
    constructor() {
        this.colorOne = '#ddd';
        this.colorTwo = '#3c8dbc';
    }
}
SlidingBarComponent.ɵfac = function SlidingBarComponent_Factory(t) { return new (t || SlidingBarComponent)(); };
SlidingBarComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SlidingBarComponent, selectors: [["mk-sliding-bar"]], inputs: { colorOne: "colorOne", colorTwo: "colorTwo" }, decls: 2, vars: 4, consts: [[1, "sliding-bar"]], template: function SlidingBarComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelement(1, "div");
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorOne);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵstyleProp("background-color", ctx.colorTwo);
    } }, styles: [".sliding-bar[_ngcontent-%COMP%]{height:4px;overflow:hidden;position:relative;width:100%}.sliding-bar[_ngcontent-%COMP%] > div[_ngcontent-%COMP%]{animation:loading 2s linear infinite;background-color:#2980b9;content:\"\";display:block;height:4px;left:-200px;position:absolute;width:200px}@keyframes loading{0%{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"] });
SlidingBarComponent.propDecorators = {
    colorOne: [{ type: Input }],
    colorTwo: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlidingBarComponent, [{
        type: Component,
        args: [{
                selector: 'mk-sliding-bar',
                template: `<div class="sliding-bar" [style.background-color]="colorOne"><div [style.background-color]="colorTwo"></div></div>`,
                styles: [".sliding-bar{height:4px;overflow:hidden;position:relative;width:100%}.sliding-bar>div{animation:loading 2s linear infinite;background-color:#2980b9;content:\"\";display:block;height:4px;left:-200px;position:absolute;width:200px}@keyframes loading{0%{left:-200px;width:30%}50%{width:30%}70%{width:70%}80%{left:50%}95%{left:120%}to{left:100%}}"]
            }]
    }], function () { return []; }, { colorOne: [{
            type: Input
        }], colorTwo: [{
            type: Input
        }] }); })();
