// angular-loading-page/esm2015/lib/loaders/sliding-bar/sliding-bar.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SlidingBarComponent } from './sliding-bar.component';
import * as ɵngcc0 from '@angular/core';
export class SlidingBarModule {
}
SlidingBarModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SlidingBarModule });
SlidingBarModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SlidingBarModule_Factory(t) { return new (t || SlidingBarModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SlidingBarModule, { declarations: function () { return [SlidingBarComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [SlidingBarComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SlidingBarModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    SlidingBarComponent,
                ],
                exports: [
                    SlidingBarComponent
                ]
            }]
    }], null, null); })();
