// angular-admin-lte/esm2015/lib/layout/footer/footer.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FooterComponent, FooterLeftComponent, FooterRightComponent } from './footer.component';
import * as ɵngcc0 from '@angular/core';
export { FooterComponent, FooterLeftComponent, FooterRightComponent } from './footer.component';
export class FooterModule {
}
FooterModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: FooterModule });
FooterModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function FooterModule_Factory(t) { return new (t || FooterModule)(); }, imports: [[CommonModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(FooterModule, { declarations: function () { return [FooterComponent, FooterLeftComponent, FooterRightComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [FooterComponent, FooterLeftComponent, FooterRightComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FooterModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                exports: [FooterComponent, FooterLeftComponent, FooterRightComponent],
                declarations: [FooterComponent, FooterLeftComponent, FooterRightComponent]
            }]
    }], null, null); })();
