// angular-admin-lte/esm2015/lib/accordion/accordion.directive.js
import { Directive, Input, ElementRef } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class AccordionToggleDirective {
    constructor(elementRef) {
        this.elementRef = elementRef;
    }
}
AccordionToggleDirective.ɵfac = function AccordionToggleDirective_Factory(t) { return new (t || AccordionToggleDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
AccordionToggleDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: AccordionToggleDirective, selectors: [["", "mkAccordionToggle", ""]], inputs: { accordionComponent: ["mkAccordionToggle", "accordionComponent"] } });
AccordionToggleDirective.ctorParameters = () => [
    { type: ElementRef }
];
AccordionToggleDirective.propDecorators = {
    accordionComponent: [{ type: Input, args: ['mkAccordionToggle',] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AccordionToggleDirective, [{
        type: Directive,
        args: [{
                selector: '[mkAccordionToggle]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { accordionComponent: [{
            type: Input,
            args: ['mkAccordionToggle']
        }] }); })();
