// angular-admin-lte/esm2015/lib/layout/content/content.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { BreadcrumbsModule } from '../../breadcrumbs/breadcrumbs.module';
import { ContentComponent } from './content.component';
import * as ɵngcc0 from '@angular/core';
export { ContentComponent } from './content.component';
export class ContentModule {
}
ContentModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ContentModule });
ContentModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ContentModule_Factory(t) { return new (t || ContentModule)(); }, imports: [[CommonModule, RouterModule, BreadcrumbsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ContentModule, { declarations: function () { return [ContentComponent]; }, imports: function () { return [CommonModule, RouterModule, BreadcrumbsModule]; }, exports: function () { return [ContentComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ContentModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule, BreadcrumbsModule],
                exports: [ContentComponent],
                declarations: [ContentComponent]
            }]
    }], null, null); })();
