// angular-admin-lte/esm2015/lib/form/input-group/input-group.module.js
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { ColorModule } from '../../color/color.module';
import { InputGroupComponent } from './input-group.component';
import { InputGroupAddonLeftDirective, InputGroupAddonRightDirective, InputGroupContentDirective, InputGroupLabelDirective } from './input-group.directive';
import * as ɵngcc0 from '@angular/core';
export { InputGroupComponent } from './input-group.component';
export { InputGroupAddonLeftDirective, InputGroupAddonRightDirective, InputGroupContentDirective, InputGroupLabelDirective } from './input-group.directive';
export class InputGroupModule {
}
InputGroupModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: InputGroupModule });
InputGroupModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function InputGroupModule_Factory(t) { return new (t || InputGroupModule)(); }, imports: [[
            CommonModule,
            ColorModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(InputGroupModule, { declarations: function () { return [InputGroupComponent, InputGroupLabelDirective, InputGroupAddonLeftDirective,
        InputGroupAddonRightDirective, InputGroupContentDirective]; }, imports: function () { return [CommonModule,
        ColorModule,
        FormsModule]; }, exports: function () { return [InputGroupComponent, InputGroupLabelDirective, InputGroupAddonLeftDirective,
        InputGroupAddonRightDirective, InputGroupContentDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputGroupModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    ColorModule,
                    FormsModule
                ],
                exports: [InputGroupComponent, InputGroupLabelDirective, InputGroupAddonLeftDirective,
                    InputGroupAddonRightDirective, InputGroupContentDirective],
                declarations: [InputGroupComponent, InputGroupLabelDirective, InputGroupAddonLeftDirective,
                    InputGroupAddonRightDirective, InputGroupContentDirective]
            }]
    }], null, null); })();
