// angular-admin-lte/esm2015/lib/form/input-group/input-group.directive.js
import { Directive } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
export class InputGroupLabelDirective {
}
InputGroupLabelDirective.ɵfac = function InputGroupLabelDirective_Factory(t) { return new (t || InputGroupLabelDirective)(); };
InputGroupLabelDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InputGroupLabelDirective, selectors: [["mk-input-group-label"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputGroupLabelDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-input-group-label'
            }]
    }], null, null); })();
export class InputGroupAddonLeftDirective {
}
InputGroupAddonLeftDirective.ɵfac = function InputGroupAddonLeftDirective_Factory(t) { return new (t || InputGroupAddonLeftDirective)(); };
InputGroupAddonLeftDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InputGroupAddonLeftDirective, selectors: [["mk-input-group-addon-left"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputGroupAddonLeftDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-input-group-addon-left'
            }]
    }], null, null); })();
export class InputGroupAddonRightDirective {
}
InputGroupAddonRightDirective.ɵfac = function InputGroupAddonRightDirective_Factory(t) { return new (t || InputGroupAddonRightDirective)(); };
InputGroupAddonRightDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InputGroupAddonRightDirective, selectors: [["mk-input-group-addon-right"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputGroupAddonRightDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-input-group-addon-right'
            }]
    }], null, null); })();
export class InputGroupContentDirective {
}
InputGroupContentDirective.ɵfac = function InputGroupContentDirective_Factory(t) { return new (t || InputGroupContentDirective)(); };
InputGroupContentDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: InputGroupContentDirective, selectors: [["mk-input-group-content"]] });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(InputGroupContentDirective, [{
        type: Directive,
        args: [{
                /* tslint:disable-next-line:directive-selector */
                selector: 'mk-input-group-content'
            }]
    }], null, null); })();
